<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Tools for Making Genetic Map Construction Easier • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" sizes="any" href="favicon.ico">
<link rel="manifest" href="site.webmanifest">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Tools for Making Genetic Map Construction Easier">
<meta name="description" content="Genetic map construction in experimental crossess. Works for F2 mapping populations with plans for extending to Backcross and Recombinant Inbred Lines.">
<meta property="og:description" content="Genetic map construction in experimental crossess. Works for F2 mapping populations with plans for extending to Backcross and Recombinant Inbred Lines.">
<meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/geneticMapR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/LD_binning.html">Binning SNP Markers by Linkage Disequilibrium</a></li>
    <li><a class="dropdown-item" href="articles/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="articles/Recode.html">Recode Markers</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="geneticmapr-">geneticMapR <a class="anchor" aria-label="anchor" href="#geneticmapr-"></a>
</h1>
</div>
<!-- badges: start -->
<!-- badges: end -->
<p><code>geneticMapR</code> simplifies the construction of genetic maps. Built on top of <a href="https://github.com/jendelman/MapRtools" class="external-link">MapRtools</a> and <a href="https://rqtl.org/" class="external-link">R/qtl</a> for reproducible analysis. Optimized for F2 diploid plant populations. Some functions were generalized for different experimental populations and <strong>polyploids</strong>. <code>geneticMapR</code> was designed to help with reproducible genetic map construction and quantitative trait loci (QTL) analysis.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of <code>geneticMapR</code> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get devtools if needed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">#Install geneticMapR</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"vegaalfaro/geneticMapR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegaalfaro/geneticMapR" class="external-link">geneticMapR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="learn-geneticmapr">Learn geneticMapR<a class="anchor" aria-label="anchor" href="#learn-geneticmapr"></a>
</h2>
<p>Check out the Articles tab to learn in detail how to construct genetic maps for F<sub>2</sub> mapping populations. Other populations under development. The vignettes (articles) are systematically designed to teach you how to create genetic maps and learn the functions along the way. <a href="https://vegaalfaro.github.io/geneticMapR/reference/index.html">geneticMapR</a> documentation also provides some examples.</p>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>The example shows how to <em>recode</em> markers when neither parent is the reference genome, as is common with <a href="https://en.wikipedia.org/wiki/Genotyping_by_sequencing" class="external-link">GBS</a> data. Molecular markers (SNPs) must be phased to accurately trace allele inheritance in the progeny. <code>recode</code> is a general function that can help with that:</p>
<div class="section level4">
<h4 id="recode-markers">Recode markers<a class="anchor" aria-label="anchor" href="#recode-markers"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"simulated_geno"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check markers previous to recoding</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">simulated_geno</span><span class="op">)</span></span>
<span><span class="co">#&gt;         Parent1 Parent2 F2_1 F2_2 F2_3</span></span>
<span><span class="co">#&gt; Marker1       0       2    0    1    2</span></span>
<span><span class="co">#&gt; Marker2       2       0    2    0    1</span></span>
<span><span class="co">#&gt; Marker3       0       2    1    2    1</span></span>
<span><span class="co">#&gt; Marker4       2       0    2    0    0</span></span>
<span><span class="co">#&gt; Marker5       0       2    0    2    1</span></span>
<span><span class="co">#&gt; Marker6       2       0    2    0    0</span></span>
<span></span>
<span><span class="co"># Recode the markers using the recode() function</span></span>
<span><span class="va">phased</span> <span class="op">&lt;-</span> <span class="fu">geneticMapR</span><span class="fu">::</span><span class="fu">recode</span><span class="op">(</span><span class="va">simulated_geno</span>, parent1 <span class="op">=</span> <span class="st">"Parent1"</span>, parent2 <span class="op">=</span> <span class="st">"Parent2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">phased</span><span class="op">)</span></span>
<span><span class="co">#&gt;         Parent1 Parent2 F2_1 F2_2 F2_3</span></span>
<span><span class="co">#&gt; Marker1       0       2    0    1    2</span></span>
<span><span class="co">#&gt; Marker2       0       2    0    2    1</span></span>
<span><span class="co">#&gt; Marker3       0       2    1    2    1</span></span>
<span><span class="co">#&gt; Marker4       0       2    0    2    2</span></span>
<span><span class="co">#&gt; Marker5       0       2    0    2    1</span></span>
<span><span class="co">#&gt; Marker6       0       2    0    2    2</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualization-capabilities">Visualization capabilities<a class="anchor" aria-label="anchor" href="#visualization-capabilities"></a>
</h4>
<p>This function generates a QTL trace plot to visualize significance scores across chromosomes for one or more traits. It highlights QTL peaks and overlays customizable vertical lines</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegaalfaro/geneticMapR" class="external-link">geneticMapR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"qtl_example"</span><span class="op">)</span></span>
<span></span>
<span>     <span class="fu"><a href="reference/plot_qtl_trace.html">plot_qtl_trace</a></span><span class="op">(</span>qtl_df <span class="op">=</span> <span class="va">qtl_example</span><span class="op">$</span><span class="va">qtl_df</span>,</span>
<span>                thresholds_df <span class="op">=</span> <span class="va">qtl_example</span><span class="op">$</span><span class="va">thresholds</span>,</span>
<span>                  vline_df <span class="op">=</span> <span class="va">qtl_example</span><span class="op">$</span><span class="va">vline</span>,</span>
<span>                  use_physical_pos <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  x_angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                trait_colors <span class="op">=</span> <span class="va">qtl_example</span><span class="op">$</span><span class="va">colors</span>,</span>
<span>                trait_labels <span class="op">=</span> <span class="va">qtl_example</span><span class="op">$</span><span class="va">labels</span>,</span>
<span>                plot_title <span class="op">=</span> <span class="st">"Population 1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-3-1.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>For detailed documentation and examples, visit <a href="https://vegaalfaro.github.io/geneticMapR/reference/index.html">geneticMapR</a></p>
</div>
<div class="section level2">
<h2 id="hexlogo">Hexlogo<a class="anchor" aria-label="anchor" href="#hexlogo"></a>
</h2>
<p>Credit: Adriana Lopez</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/vegaalfaro/geneticMapR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/vegaalfaro/geneticMapR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing geneticMapR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Andrey Vega Alfaro <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3293-9456" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
