<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Filter Genotype Matrix by Allele Frequency — filter_geno_by_freq • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Filter Genotype Matrix by Allele Frequency — filter_geno_by_freq"><meta name="description" content="Filters a genotype matrix by applying constraints on the maximum
and minimum genotype frequency per marker, as well as heterozygous frequency.
The filtering is based on allele frequency calculations using the freq() function internally."><meta property="og:description" content="Filters a genotype matrix by applying constraints on the maximum
and minimum genotype frequency per marker, as well as heterozygous frequency.
The filtering is based on allele frequency calculations using the freq() function internally."><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Haplotype_and_Map.html">Genetic Map</a></li>
    <li><a class="dropdown-item" href="../articles/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../articles/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../articles/pheno_part_2.html">Phenotypic Data Part II</a></li>
    <li><a class="dropdown-item" href="../articles/pheno.html">Phenotypic Data Part I</a></li>
    <li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Filter Genotype Matrix by Allele Frequency</h1>
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/R/filter_geno_by_freq.R" class="external-link"><code>R/filter_geno_by_freq.R</code></a></small>
      <div class="d-none name"><code>filter_geno_by_freq.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Filters a genotype matrix by applying constraints on the maximum
and minimum genotype frequency per marker, as well as heterozygous frequency.
The filtering is based on allele frequency calculations using the <code><a href="freq.html">freq()</a></code> function internally.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">filter_geno_by_freq</span><span class="op">(</span></span>
<span>  <span class="va">geno_matrix</span>,</span>
<span>  max_geno_freq <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  het_freq_range <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_geno_freq <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  input_format <span class="op">=</span> <span class="st">"numeric"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-geno-matrix">geno_matrix<a class="anchor" aria-label="anchor" href="#arg-geno-matrix"></a></dt>
<dd><p>A numeric genotype matrix or data frame where:</p><ul><li><p>Rows represent genetic markers.</p></li>
<li><p>Columns represent individuals.</p></li>
<li><p>Values are either <code>0, 1, 2</code> (numeric format) or <code>"A", "H", "B"</code> (genotype format).</p></li>
</ul></dd>


<dt id="arg-max-geno-freq">max_geno_freq<a class="anchor" aria-label="anchor" href="#arg-max-geno-freq"></a></dt>
<dd><p>Numeric. If provided, removes markers where the most frequent
genotype exceeds this threshold.</p></dd>


<dt id="arg-het-freq-range">het_freq_range<a class="anchor" aria-label="anchor" href="#arg-het-freq-range"></a></dt>
<dd><p>Numeric vector of length 2. If provided, retains markers where
heterozygosity frequency is within the specified range (<code>c(min, max)</code>).</p></dd>


<dt id="arg-min-geno-freq">min_geno_freq<a class="anchor" aria-label="anchor" href="#arg-min-geno-freq"></a></dt>
<dd><p>Numeric. If provided, removes markers where the least frequent
genotype falls below this threshold.</p></dd>


<dt id="arg-input-format">input_format<a class="anchor" aria-label="anchor" href="#arg-input-format"></a></dt>
<dd><p>Character. Specifies whether the genotype matrix is in <code>"numeric"</code>
(<code>0, 1, 2</code>) or <code>"genotype"</code> (<code>"A", "H", "B"</code>) format. Default is <code>"numeric"</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A filtered genotype matrix with only markers that meet the specified
frequency criteria.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>

<ul><li><p>Computes genotype frequencies using the <code><a href="freq.html">freq()</a></code> function.</p></li>
<li><p>Retains only markers that meet the specified frequency constraints.</p></li>
<li><p>If all filtering parameters (<code>max_geno_freq</code>, <code>het_freq_range</code>, <code>min_geno_freq</code>) are <code>NULL</code>,
the function returns the original matrix.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example genotype matrix</span></span></span>
<span class="r-in"><span><span class="va">geno_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">30</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    nrow <span class="op">=</span> <span class="fl">10</span>, ncol <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>                    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Marker"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Ind"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filter markers with max genotype frequency &lt; 0.95, heterozygosity between 0.1 and 0.8,</span></span></span>
<span class="r-in"><span><span class="co"># and minimum genotype frequency &gt;= 0.05</span></span></span>
<span class="r-in"><span><span class="va">filtered_data</span> <span class="op">&lt;-</span> <span class="fu">filter_geno_by_freq</span><span class="op">(</span><span class="va">geno_data</span>, max_geno_freq <span class="op">=</span> <span class="fl">0.95</span>,</span></span>
<span class="r-in"><span>                                     het_freq_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.80</span><span class="op">)</span>, min_geno_freq <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

