<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>F2 diploid population Haplotype Reconstruction — haplotype_reconstruction • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="F2 diploid population Haplotype Reconstruction — haplotype_reconstruction"><meta name="description" content="This function performs haplotype reconstruction using a Hidden Markov Model (HMM).
It applies the Viterbi algorithm to infer the most likely sequence of true genotypic
states, accounting for genotyping errors and missing data."><meta property="og:description" content="This function performs haplotype reconstruction using a Hidden Markov Model (HMM).
It applies the Viterbi algorithm to infer the most likely sequence of true genotypic
states, accounting for genotyping errors and missing data."><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>F2 diploid population Haplotype Reconstruction</h1>
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/R/haplotype_reconstruction.R" class="external-link"><code>R/haplotype_reconstruction.R</code></a></small>
      <div class="d-none name"><code>haplotype_reconstruction.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs haplotype reconstruction using a Hidden Markov Model (HMM).
It applies the Viterbi algorithm to infer the most likely sequence of true genotypic
states, accounting for genotyping errors and missing data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">haplotype_reconstruction</span><span class="op">(</span><span class="va">geno_matrix</span>, error_rate <span class="op">=</span> <span class="fl">0.05</span>, r <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-geno-matrix">geno_matrix<a class="anchor" aria-label="anchor" href="#arg-geno-matrix"></a></dt>
<dd><p>A numeric genotype matrix where:</p><ul><li><p>Rows represent genetic markers.</p></li>
<li><p>Columns represent progeny (individuals).</p></li>
<li><p>Values are expected to be <code>0</code>, <code>1</code>, <code>2</code>, or <code>NA</code> for missing data.</p></li>
</ul></dd>


<dt id="arg-error-rate">error_rate<a class="anchor" aria-label="anchor" href="#arg-error-rate"></a></dt>
<dd><p>Numeric. The assumed genotyping error rate. Default is <code>0.05</code> (5% error rate).</p></dd>


<dt id="arg-r">r<a class="anchor" aria-label="anchor" href="#arg-r"></a></dt>
<dd><p>Numeric. The recombination rate used in the transition probability matrix (<code>T.mat</code>).
Default is <code>0.01</code>. The user can use the <code>MLEL</code> function from <a href="https://github.com/jendelman/MapRtools" class="external-link">MapRtools</a>
to get an estimate of recombination frequency with the adjacent argument set to <code>TRUE</code>
<code>result &lt;- MLEL(geno = geno_matrix, pop.type = "f2", LOD = FALSE, adjacent = TRUE)</code>
In this case the geno_matrix should be of only one chromosome. <code>r</code> could be estimated as <code>mean(result$value, na.rm = TRUE)</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A genotype matrix with inferred haplotypes based on HMM correction.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>

<ul><li><p>Estimates the missing data rate in the input genotype matrix.</p></li>
<li><p>Initializes an HMM using:</p><ul><li><p>Three hidden states (<code>"0"</code>, <code>"1"</code>, <code>"2"</code>).</p></li>
<li><p>Four observable symbols (<code>"0"</code>, <code>"1"</code>, <code>"2"</code>, <code>"NA"</code> for missing data).</p></li>
<li><p>Transition probabilities generated via <code>T.mat(r = r)</code>, using the user-defined recombination rate.</p></li>
<li><p>Emission probabilities computed using <code>E.mat(error = error_rate, missing = missing)</code>.</p></li>
</ul></li>
</ul></div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Works only for F2 populations on experimental crosses. Based on the 615 Genetic Mapping Class notes by
Prof. Jeffrey Endelman. Spring 2021.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Example genotype matrix with missing values</span></span></span>
<span class="r-in"><span><span class="va">geno_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="cn">NA</span>, <span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="cn">NA</span>, <span class="fl">2</span>, <span class="cn">NA</span>, <span class="cn">NA</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    nrow <span class="op">=</span> <span class="fl">4</span>, ncol <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>                    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Marker1"</span>, <span class="st">"Marker2"</span>, <span class="st">"Marker3"</span>, <span class="st">"Marker4"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ind1"</span>, <span class="st">"Ind2"</span>, <span class="st">"Ind3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Perform haplotype reconstruction with default parameters</span></span></span>
<span class="r-in"><span><span class="va">reconstructed_geno</span> <span class="op">&lt;-</span> <span class="fu">haplotype_reconstruction</span><span class="op">(</span><span class="va">geno_data</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">reconstructed_geno</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Perform haplotype reconstruction with a modified recombination rate</span></span></span>
<span class="r-in"><span><span class="va">reconstructed_geno_custom_r</span> <span class="op">&lt;-</span> <span class="fu">haplotype_reconstruction</span><span class="op">(</span><span class="va">geno_data</span>, error_rate <span class="op">=</span> <span class="fl">0.05</span>, r <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">reconstructed_geno_custom_r</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

