<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Haplotype_and_Map.html">Genetic Map</a></li>
    <li><a class="dropdown-item" href="../articles/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../articles/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../articles/pheno.html">Phenotypic Data Part I</a></li>
    <li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="core-functions">Core Functions<a class="anchor" aria-label="anchor" href="#core-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="convert_to_dosage_advanced.html">convert_to_dosage_advanced()</a></code>

        </dt>
        <dd>Convert VCF Genotype Format to Dosage (Advanced, Polyploid-Compatible)</dd>
      </dl><dl><dt>

          <code><a href="convert_to_dosage.html">convert_to_dosage()</a></code>

        </dt>
        <dd>Convert VCF Genotype Format to Dosage for Diploids</dd>
      </dl><dl><dt>

          <code><a href="convert_to_dosage_flex.html">convert_to_dosage_flex()</a></code>

        </dt>
        <dd>Convert VCF Genotype Format to Dosage (Supports Polyploids and Missing Alleles)</dd>
      </dl><dl><dt>

          <code><a href="effect_plot.html">effect_plot()</a></code>

        </dt>
        <dd>Powerful Effect Plots</dd>
      </dl><dl><dt>

          <code><a href="estimate_map.html">estimate_map()</a></code>

        </dt>
        <dd>Estimate Genetic Map Positions</dd>
      </dl><dl><dt>

          <code><a href="extract_map.html">extract_map()</a></code>

        </dt>
        <dd>Create a Physical Map from Chromosome and Position from Row Names</dd>
      </dl><dl><dt>

          <code><a href="freq.html">freq()</a></code>

        </dt>
        <dd>Calculate Genotype Frequency Per Individual or Marker</dd>
      </dl><dl><dt>

          <code><a href="get_LOD_int.html">get_LOD_int()</a></code>

        </dt>
        <dd>Get LOD Support Intervals for QTL Peaks (Single or Multiple Traits)</dd>
      </dl><dl><dt>

          <code><a href="haplotype_reconstruction.html">haplotype_reconstruction()</a></code>

        </dt>
        <dd>F2 diploid population Haplotype Reconstruction</dd>
      </dl><dl><dt>

          <code><a href="recode_markers.html">recode_markers()</a></code>

        </dt>
        <dd>Phase Genotype Marker Data Based on Parental References</dd>
      </dl><dl><dt>

          <code><a href="simple_effect_plot.html">simple_effect_plot()</a></code>

        </dt>
        <dd>Simple Effect Plot</dd>
      </dl><dl><dt>

          <code><a href="trim_LG.html">trim_LG()</a></code>

        </dt>
        <dd>Trim Linkage Groups Based on LOD Thresholds</dd>
      </dl></div><div class="section level2">
      <h2 id="filtering-formatting-and-plotting">Filtering, Formatting and Plotting<a class="anchor" aria-label="anchor" href="#filtering-formatting-and-plotting"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="filter_geno_by_freq.html">filter_geno_by_freq()</a></code>

        </dt>
        <dd>Filter Genotype Matrix by Allele Frequency</dd>
      </dl><dl><dt>

          <code><a href="filter_geno_by_freq_poly.html">filter_geno_by_freq_poly()</a></code>

        </dt>
        <dd>Filter Genotype Matrix by Dosage Frequency (Polyploid-Compatible)</dd>
      </dl><dl><dt>

          <code><a href="filter_geno_by_parents.html">filter_geno_by_parents()</a></code>

        </dt>
        <dd>Filter Genotype Matrix Based on Parental Genotype</dd>
      </dl><dl><dt>

          <code><a href="filter_missing_geno.html">filter_missing_geno()</a></code>

        </dt>
        <dd>Filter Markers or Individuals Based on Missing Genotype Data</dd>
      </dl><dl><dt>

          <code><a href="formater.html">formater()</a></code>

        </dt>
        <dd>Convert Between Character (A, H, B) and Numeric dosage (0, 1, 2) Formats</dd>
      </dl><dl><dt>

          <code><a href="formater_flex.html">formater_flex()</a></code>

        </dt>
        <dd>Flexible Genotype Format Converter (Supports Polyploid Dosage)</dd>
      </dl><dl><dt>

          <code><a href="format_qtl_input.html">format_qtl_input()</a></code>

        </dt>
        <dd>Format Input Data for R/qtl package</dd>
      </dl><dl><dt>

          <code><a href="LD_decay_plot.html">LD_decay_plot()</a></code>

        </dt>
        <dd>Plot LD vs distance</dd>
      </dl><dl><dt>

          <code><a href="frequency_plot.html">frequency_plot()</a></code>

        </dt>
        <dd>Generate a Histogram of Genotype Frequencies</dd>
      </dl><dl><dt>

          <code><a href="order_and_plot.html">order_and_plot()</a></code>

        </dt>
        <dd>Optimize Marker Order</dd>
      </dl><dl><dt>

          <code><a href="plotMap_gg.html">plotMap_gg()</a></code>

        </dt>
        <dd>Plot Genetic Map Using ggplot2 in R/qtl data environment</dd>
      </dl><dl><dt>

          <code><a href="plot_qtl_trace.html">plot_qtl_trace()</a></code>

        </dt>
        <dd>Generates a line plot showing QTL traces (LOD scores) across chromosomes for one or more traits. Supports customization of visual appearance and annotations such as significance thresholds and vertical lines.</dd>
      </dl><dl><dt>

          <code><a href="plot_cover.html">plot_cover()</a></code>

        </dt>
        <dd>Plot Genome-wide Coverage</dd>
      </dl><dl><dt>

          <code><a href="plot_coverage_annotate.html">plot_coverage_annotate()</a></code>

        </dt>
        <dd>Plot Coverage Map with Candidate Gene Annotations</dd>
      </dl><dl><dt>

          <code><a href="plot_genotype_histogram.html">plot_genotype_histogram()</a></code>

        </dt>
        <dd>Plot a Histogram of Genotypic Values</dd>
      </dl><dl><dt>

          <code><a href="plot_marey.html">plot_marey()</a></code>

        </dt>
        <dd>Generate a Marey Plot for one Chromosome</dd>
      </dl><dl><dt>

          <code><a href="plot_reconstructed_haplo.html">plot_reconstructed_haplo()</a></code>

        </dt>
        <dd>Plot Haplotype Reconstruction Results Before and After HMM</dd>
      </dl></div><div class="section level2">
      <h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="simulated_geno.html">simulated_geno</a></code>

        </dt>
        <dd>Simulated Simple Genotype Matrix Dataset</dd>
      </dl><dl><dt>

          <code><a href="qtl_example.html">qtl_example</a></code>

        </dt>
        <dd>QTL Dataset Example</dd>
      </dl></div><div class="section level2">
      <h2 id="utility-functions">Utility Functions<a class="anchor" aria-label="anchor" href="#utility-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="drop_parents.html">drop_parents()</a></code>

        </dt>
        <dd>Drop Parents and F1 Individuals from a Data Frame</dd>
      </dl><dl><dt>

          <code><a href="melt_freq.html">melt_freq()</a></code>

        </dt>
        <dd>Convert (Melt) Genotype Frequency Data to Long Format</dd>
      </dl><dl><dt>

          <code><a href="rename_geno_matrix.html">rename_geno_matrix()</a></code>

        </dt>
        <dd>Rename Parents, F1, and F2 Individuals in a Genotype Matrix</dd>
      </dl><dl><dt>

          <code><a href="estimate_error.html">estimate_error()</a></code>

        </dt>
        <dd>Estimate Genotyping Error Rate</dd>
      </dl><dl><dt>

          <code><a href="load_or_install_cran.html">load_or_install_cran()</a></code>

        </dt>
        <dd>Load or Install a CRAN Package</dd>
      </dl><dl><dt>

          <code><a href="load_or_install_github.html">load_or_install_github()</a></code>

        </dt>
        <dd>Load or Install a GitHub Package</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

