<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot LD vs distance — LD_decay_plot • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot LD vs distance — LD_decay_plot"><meta name="description" content="Plot LD vs distance using an asymptotic regression model."><meta property="og:description" content="Plot LD vs distance using an asymptotic regression model."><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../vignettes/Recode.html">Recode Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno.html">Phenotype Part I</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno_part_2.html">Phenotype Part II</a></li>
    <li><a class="dropdown-item" href="../vignettes/qtl_analysis.html">Linkage Groups and Marker Order</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Plot LD vs distance</h1>
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/R/LD_decay_plot.R" class="external-link"><code>R/LD_decay_plot.R</code></a></small>
      <div class="d-none name"><code>LD_decay_plot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plot LD vs distance using an asymptotic regression model.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">LD_decay_plot</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">map</span>,</span>
<span>  max.pair <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  max.loci <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"bp"</span>,</span>
<span>  r2_threshold <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  xlim_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">20</span><span class="op">)</span>,</span>
<span>  show_vline <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_hline <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">123</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>Genotype matrix (markers as columns, individuals as rows). Rownames have the marker names that matcho those of the marker column in the <code>map</code>.</p></dd>


<dt id="arg-map">map<a class="anchor" aria-label="anchor" href="#arg-map"></a></dt>
<dd><p>Data frame with columns 'chrom', 'marker', and 'position'</p></dd>


<dt id="arg-max-pair">max.pair<a class="anchor" aria-label="anchor" href="#arg-max-pair"></a></dt>
<dd><p>Maximum number of r² pairs for the model (default: 1e4)</p></dd>


<dt id="arg-max-loci">max.loci<a class="anchor" aria-label="anchor" href="#arg-max-loci"></a></dt>
<dd><p>Maximum number of markers to use per chromosome (default: NULL)</p></dd>


<dt id="arg-position">position<a class="anchor" aria-label="anchor" href="#arg-position"></a></dt>
<dd><p>"bp" or "Mb" (default: "bp")</p></dd>


<dt id="arg-r-threshold">r2_threshold<a class="anchor" aria-label="anchor" href="#arg-r-threshold"></a></dt>
<dd><p>r² value for decay distance calculation (default: 0.2),</p></dd>


<dt id="arg-xlim-range">xlim_range<a class="anchor" aria-label="anchor" href="#arg-xlim-range"></a></dt>
<dd><p>Zoom in range on the x axis, default is c(0, 20). Set to NULL to see entire plot.</p></dd>


<dt id="arg-show-vline">show_vline<a class="anchor" aria-label="anchor" href="#arg-show-vline"></a></dt>
<dd><p>show a vertical line intersecting the x-axis at the the half-decay distance (default, TRUE)</p></dd>


<dt id="arg-show-hline">show_hline<a class="anchor" aria-label="anchor" href="#arg-show-hline"></a></dt>
<dd><p>show a horizontal line intersecting the y-axis at the selected r² threshold (default, TRUE)</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>Optional integer seed to make the random sampling reproducible. Default 123.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing:</p>
<dl><dt>plot</dt>
<dd><p>ggplot2 object with points and fitted curve (or NULL if fit fails)</p></dd>

<dt>half_decay_dist</dt>
<dd><p>Distance where r² reaches the specified threshold (or NA if not estimable)</p></dd>

<dt>model</dt>
<dd><p>Fitted nls object (or NULL if fitting failed)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>An asymptotic regression model is fit using <a href="https://rdrr.io/r/stats/SSasymp.html" class="external-link">SSasymp</a>.
The distance where r² reaches a specified threshold (e.g., 0.1 or 0.2) is calculated and returned.
This function calculates r² by chromosome and  follows the general logic of <a href="https://github.com/jendelman/MapRtools/blob/master/R/plot_LD.R" class="external-link">MapRtools::LD.plot</a> but uses the function
<a href="https://rdrr.io/r/stats/SSasymp.html" class="external-link">SSasymp</a> to model the non-linear LD decay. This function will give you a
quick and simple estimate of LD. For more sophisticated LD functions see <a href="https://github.com/dcgerard" class="external-link">David Gerard</a> <a href="https://cran.r-project.org/web/packages/ldsep/vignettes/fast.html" class="external-link">ldfast()</a></p>
<p>It processes each chromosome separately to compute all pairwise r²
values within chromosomes, then it combines the results across all
chromosomes into a single dataset, and fits one global LD decay model
to the pooled data. No cross-chromosome marker pairs are considered.</p>
<p>If the user is interested in LD per chromosome, provide a map and data file with only
data pertaining to the chromosome of interest.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>If the number of rows (number of marker pairs) is larger than max.pair (default is 1e4), we proceed to random sampling.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

