<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Phase Genotype Marker Data Based on Parental References — recode_markers • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Phase Genotype Marker Data Based on Parental References — recode_markers"><meta name="description" content='recode_markers is the heart of the package. This powerful function phases genotype marker
data based on two parental references (parent1 and parent2).
It phases markers according to parental allele inheritance.
Optionally, it can code phased markers into numeric (0, 1, 2) or character ("A", "B", "H") formats.
Numeric coding is recommended for downstream analyses.'><meta property="og:description" content='recode_markers is the heart of the package. This powerful function phases genotype marker
data based on two parental references (parent1 and parent2).
It phases markers according to parental allele inheritance.
Optionally, it can code phased markers into numeric (0, 1, 2) or character ("A", "B", "H") formats.
Numeric coding is recommended for downstream analyses.'><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Haplotype_and_Map.html">Genetic Map</a></li>
    <li><a class="dropdown-item" href="../articles/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../articles/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../articles/pheno.html">Phenotypic Data Part I</a></li>
    <li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Phase Genotype Marker Data Based on Parental References</h1>
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/R/recode_markers.R" class="external-link"><code>R/recode_markers.R</code></a></small>
      <div class="d-none name"><code>recode_markers.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>recode_markers</code> is the heart of the package. This powerful function phases genotype marker
data based on two parental references (<code>parent1</code> and <code>parent2</code>).</p>
<p>It phases markers according to parental allele inheritance.
Optionally, it can code phased markers into numeric (<code>0</code>, <code>1</code>, <code>2</code>) or character (<code>"A"</code>, <code>"B"</code>, <code>"H"</code>) formats.
Numeric coding is recommended for downstream analyses.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">recode_markers</span><span class="op">(</span></span>
<span>  <span class="va">geno</span>,</span>
<span>  <span class="va">parent1</span>,</span>
<span>  <span class="va">parent2</span>,</span>
<span>  numeric_output <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  handle_het_markers <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  het_marker_types <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-geno">geno<a class="anchor" aria-label="anchor" href="#arg-geno"></a></dt>
<dd><p>A genotype matrix or data frame where markers are rows and individuals are columns.</p></dd>


<dt id="arg-parent-">parent1<a class="anchor" aria-label="anchor" href="#arg-parent-"></a></dt>
<dd><p>Character. The name of the column representing the first parent.</p></dd>


<dt id="arg-parent-">parent2<a class="anchor" aria-label="anchor" href="#arg-parent-"></a></dt>
<dd><p>Character. The name of the column representing the second parent.</p></dd>


<dt id="arg-numeric-output">numeric_output<a class="anchor" aria-label="anchor" href="#arg-numeric-output"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, converts phased markers to numeric dosage values
(<code>A = 0</code>, <code>H = 1</code>, <code>B = 2</code>). Default is <code>TRUE</code>.</p></dd>


<dt id="arg-handle-het-markers">handle_het_markers<a class="anchor" aria-label="anchor" href="#arg-handle-het-markers"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, allows heterozygous parent markers to be included.
Default is <code>FALSE</code>.</p></dd>


<dt id="arg-het-marker-types">het_marker_types<a class="anchor" aria-label="anchor" href="#arg-het-marker-types"></a></dt>
<dd><p>Character vector. Specifies which heterozygous markers to keep
when <code>handle_het_markers = TRUE</code>. Options include <code>"AxH"</code>, <code>"HxB"</code>, <code>"HxA"</code>, <code>"BxH"</code>.
Default is <code>NULL</code>, meaning all homozygous markers are kept. See details.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame containing phased genotype markers where:</p><ul><li><p><code>"0"</code> represents alleles inherited from <code>parent1</code>.</p></li>
<li><p><code>"2"</code> represents alleles inherited from <code>parent2</code>.</p></li>
<li><p><code>"1"</code> represents heterozygous alleles.</p></li>
<li><p>If <code>numeric_output = FALSE</code>,   <code>0</code>, <code>2</code>, <code>1</code>, are replaced by <code>"A"</code>, <code>"B"</code>, <code>"H"</code> respectively.</p></li>
<li><p><code>numeric_output = TRUE</code> is recommended.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>

<ul><li><p>Drops markers where either parent has <code>NA</code>.</p></li>
<li><p>Removes non-polymorphic markers (markers where both parents have the same genotype).</p></li>
<li><p>If <code>handle_het_markers = FALSE</code>, retains only homozygous marker where dosages are as follows: (<code>P1 = 0</code> &amp; <code>P2 = 2</code> or <code>P1 = 2</code> &amp; <code>P2 = 0</code>)</p></li>
<li><p>If <code>handle_het_markers = TRUE</code>, allows heterozygous markers to be kept.</p></li>
<li><p>Ensures that <code>parent1</code> is always <code>0</code> and <code>parent2</code> is always <code>2</code> for standardization.</p></li>
<li><p>Returns a numeric matrix if <code>numeric_output = TRUE</code>, otherwise returns phased <code>"A"</code>, <code>"B"</code>, <code>"H"</code> values.</p></li>
<li><p>More details on the heterozygous F2 marker types ("AxH", "HxB", "HxA", "BxH") are in <a href="https://acsess.onlinelibrary.wiley.com/doi/10.3835/plantgenome2016.10.0110" class="external-link">Braun et al. (2017).</a></p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Example genotype data</span></span></span>
<span class="r-in"><span><span class="va">geno_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  Marker1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  Marker2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  Parent1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  Parent2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># recode_markers genotype markers (default: numeric output)</span></span></span>
<span class="r-in"><span><span class="va">phased_geno</span> <span class="op">&lt;-</span> <span class="fu">recode_markers</span><span class="op">(</span><span class="va">geno_data</span>, <span class="st">"Parent1"</span>, <span class="st">"Parent2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">phased_geno</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Marker1 Marker2 Parent1 Parent2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       0       2       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2       1       2       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3       0       0       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4       0       1       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5       0       2       0       2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Recode genotype markers with heterozygous marker handling</span></span></span>
<span class="r-in"><span><span class="va">phased_geno_het</span> <span class="op">&lt;-</span> <span class="fu">recode_markers</span><span class="op">(</span><span class="va">geno_data</span>, <span class="st">"Parent1"</span>, <span class="st">"Parent2"</span>,</span></span>
<span class="r-in"><span>                          handle_het_markers <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                          het_marker_types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AxH"</span>, <span class="st">"HxB"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">phased_geno_het</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Marker1 Marker2 Parent1 Parent2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1       0       2       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2       1       2       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3       0       0       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4       0       1       0       2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5       0       2       0       2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

