<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Simple Effect Plot — simple_effect_plot • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Simple Effect Plot — simple_effect_plot"><meta name="description" content="Generates a clean and simple effect plot. Shows the relationship between genotype classes at a marker on a trait.
Optionally includes annotation of medians and sample sizes, and allows flipping of axes.
Works on polyploids."><meta property="og:description" content="Generates a clean and simple effect plot. Shows the relationship between genotype classes at a marker on a trait.
Optionally includes annotation of medians and sample sizes, and allows flipping of axes.
Works on polyploids."><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../vignettes/Recode.html">Recode Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno.html">Phenotype Part I</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno_part_2.html">Phenotype Part II</a></li>
    <li><a class="dropdown-item" href="../vignettes/qtl_analysis.html">Linkage Groups and Marker Order</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Simple Effect Plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/R/simple_effect_plot.R" class="external-link"><code>R/simple_effect_plot.R</code></a></small>
      <div class="d-none name"><code>simple_effect_plot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Generates a clean and simple effect plot. Shows the relationship between genotype classes at a marker on a trait.
Optionally includes annotation of medians and sample sizes, and allows flipping of axes.
<strong>Works on polyploids.</strong></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">simple_effect_plot</span><span class="op">(</span></span>
<span>  <span class="va">effects_df</span>,</span>
<span>  <span class="va">marker_name</span>,</span>
<span>  <span class="va">trait_name</span>,</span>
<span>  genotype_levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"H"</span>, <span class="st">"B"</span><span class="op">)</span>,</span>
<span>  annotate <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  flip <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  trait_label <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-effects-df">effects_df<a class="anchor" aria-label="anchor" href="#arg-effects-df"></a></dt>
<dd><p>A data frame containing at least the columns for marker genotypes and trait values. Usually the output of <code>format_qtl_input</code>with a few modifications (first two rows removed).</p></dd>


<dt id="arg-marker-name">marker_name<a class="anchor" aria-label="anchor" href="#arg-marker-name"></a></dt>
<dd><p>A character string specifying the name of the marker column in <code>effects_df</code>.</p></dd>


<dt id="arg-trait-name">trait_name<a class="anchor" aria-label="anchor" href="#arg-trait-name"></a></dt>
<dd><p>A character string specifying the name of the trait column in <code>effects_df</code>.</p></dd>


<dt id="arg-genotype-levels">genotype_levels<a class="anchor" aria-label="anchor" href="#arg-genotype-levels"></a></dt>
<dd><p>A character vector indicating the expected genotype categories (factor levels). Default is <code>c("A", "H", "B")</code>.</p></dd>


<dt id="arg-annotate">annotate<a class="anchor" aria-label="anchor" href="#arg-annotate"></a></dt>
<dd><p>Logical; if <code>TRUE</code>, the plot includes annotations of median values and sample sizes per genotype. Default is <code>TRUE</code>.</p></dd>


<dt id="arg-flip">flip<a class="anchor" aria-label="anchor" href="#arg-flip"></a></dt>
<dd><p>Logical; if <code>TRUE</code>, the coordinates of the plot are flipped (horizontal layout). Default is <code>FALSE</code>.</p></dd>


<dt id="arg-trait-label">trait_label<a class="anchor" aria-label="anchor" href="#arg-trait-label"></a></dt>
<dd><p>Optional character string to use as the y-axis label. If <code>NULL</code>, the trait name is used.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code>ggplot2</code> object displaying the distribution of trait values across genotype classes (alleles) at the specified marker.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function performs the following:</p><ul><li><p>Verifies the presence of marker and trait columns in the input data frame.</p></li>
<li><p>Removes rows with missing genotype data.</p></li>
<li><p>Converts the marker column to a factor with specified genotype levels.</p></li>
<li><p>Creates a boxplot with optional annotations of median values and counts.</p></li>
<li><p>Optionally flips the plot horizontally.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Inspired by  Fig. 2  <a href="https://doi.org/10.1007/s00122-022-04159-z" class="external-link">Caraza-Harter &amp; Endelman</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>

    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

