<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Trim Linkage Groups Based on LOD Thresholds — trim_LG • geneticMapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Trim Linkage Groups Based on LOD Thresholds — trim_LG"><meta name="description" content="trim_LG  is an interactive function that helps the user filter markers within a specified chromosome based on linkage
group (LG) assignment using LOD score thresholds. trim_LG  should make the process easy by allowing the user
to choose the thresholds interactively, visualize haplotype frequency before and after filtering,
and  remove outliers. Inspired by functions by Professor Jeffrey B. Endelman's MapRtools
available in Github. trim_LG provides a simple interphase for trimming LGs easily."><meta property="og:description" content="trim_LG  is an interactive function that helps the user filter markers within a specified chromosome based on linkage
group (LG) assignment using LOD score thresholds. trim_LG  should make the process easy by allowing the user
to choose the thresholds interactively, visualize haplotype frequency before and after filtering,
and  remove outliers. Inspired by functions by Professor Jeffrey B. Endelman's MapRtools
available in Github. trim_LG provides a simple interphase for trimming LGs easily."><meta property="og:image" content="https://vegaalfaro.github.io/geneticMapR/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Trim Linkage Groups Based on LOD Thresholds</h1>
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/R/trim_LG.R" class="external-link"><code>R/trim_LG.R</code></a></small>
      <div class="d-none name"><code>trim_LG.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>trim_LG</code>  is an interactive function that helps the user filter markers within a specified chromosome based on linkage
group (LG) assignment using LOD score thresholds. <code>trim_LG</code>  should make the process easy by allowing the user
to choose the thresholds interactively, visualize haplotype frequency before and after filtering,
and  remove outliers. Inspired by functions by Professor Jeffrey B. Endelman's <a href="https://github.com/jendelman/MapRtools" class="external-link">MapRtools</a>
available in Github. <code>trim_LG</code> provides a simple interphase for trimming LGs easily.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">trim_LG</span><span class="op">(</span></span>
<span>  <span class="va">chromosome</span>,</span>
<span>  <span class="va">map</span>,</span>
<span>  <span class="va">geno</span>,</span>
<span>  pop_type <span class="op">=</span> <span class="st">"F2"</span>,</span>
<span>  drop_outliers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_cores <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-chromosome">chromosome<a class="anchor" aria-label="anchor" href="#arg-chromosome"></a></dt>
<dd><p>Character. The chromosome ID to be processed.</p></dd>


<dt id="arg-map">map<a class="anchor" aria-label="anchor" href="#arg-map"></a></dt>
<dd><p>A data frame containing marker map information with at least the following columns:</p><ul><li><p><code>"marker"</code>: Marker names.</p></li>
<li><p><code>"chrom"</code>: Chromosome identifier.</p></li>
<li><p><code>"position"</code>: Physical position of markers.</p></li>
</ul></dd>


<dt id="arg-geno">geno<a class="anchor" aria-label="anchor" href="#arg-geno"></a></dt>
<dd><p>A genotype matrix where:</p><ul><li><p>Rows represent genetic markers.</p></li>
<li><p>Columns represent individuals.</p></li>
<li><p>Values represent genotype calls.
Preferably, a binned genotype matrix. Binning can be performed using <code>MapRtools::LDbin</code>.</p></li>
</ul></dd>


<dt id="arg-pop-type">pop_type<a class="anchor" aria-label="anchor" href="#arg-pop-type"></a></dt>
<dd><p>Character. The population type used for linkage estimation. Default is <code>"F2"</code>. Would work with any of the following "DH","BC","F2","S1","RIL.self","RIL.sib". Based on <code>MapRtools::MLEL()</code></p></dd>


<dt id="arg-drop-outliers">drop_outliers<a class="anchor" aria-label="anchor" href="#arg-drop-outliers"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, removes markers identified as outliers based on haplotype frequency. Default is <code>TRUE</code>.</p></dd>


<dt id="arg-n-cores">n_cores<a class="anchor" aria-label="anchor" href="#arg-n-cores"></a></dt>
<dd><p>Integer. The number of CPU cores to use for linkage estimation.
If <code>NULL</code>, the function selects the maximum available minus one.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing:</p><ul><li><p><code>"trimmed_genotype"</code>: The filtered genotype matrix.</p></li>
<li><p><code>"final_map"</code>: The updated marker map after filtering.</p></li>
<li><p><code>"initial_haplo_plot"</code>: A <code>ggplot2</code> object showing the initial haplotype plot.</p></li>
<li><p><code>"filtered_freq_plot"</code>: A <code>ggplot2</code> object showing the haplotype plot after filtering.</p></li>
<li><p><code>"final_freq_plot"</code>: A <code>ggplot2</code> object showing the haplotype frequency after removing outliers.</p></li>
<li><p><code>"starting_LOD"</code>: A record of the user-specified initial LOD threshold.</p></li>
<li><p><code>"ending_LOD"</code>: A record of the the user-specified final LOD threshold.</p></li>
<li><p><code>"step"</code>: A record of the LOD sequence step size.</p></li>
<li><p><code>"selected_LOD_thresh"</code>: The final LOD threshold used for filtering markers.</p></li>
<li><p><code>"outliers"</code>: Markers removed as outliers saved as character vector.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>

<ul><li><p>Computes LOD scores for marker pairs using <code>MapRtools::MLEL()</code>, parallelized with multiple cores.</p></li>
<li><p>Displays an interactive "candy stripe" plot for visualizing linkage groups using <code>MapRtools::LG()</code>.</p></li>
<li><p>Asks the user to define a LOD range and a final LOD threshold for marker filtering.</p></li>
<li><p>If <code>drop_outliers = TRUE</code>, removes markers identified as outliers based on haplotype frequency.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Requires a previous installation of <a href="https://github.com/jendelman/MapRtools" class="external-link">MapRtools</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Example dataset (user should provide actual data)</span></span></span>
<span class="r-in"><span><span class="va">map_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  marker <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M1"</span>, <span class="st">"M2"</span>, <span class="st">"M3"</span>, <span class="st">"M4"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  chrom <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHR1"</span>, <span class="st">"CHR1"</span>, <span class="st">"CHR1"</span>, <span class="st">"CHR1"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span>, <span class="fl">40</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">geno_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">1</span>, <span class="fl">16</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      nrow <span class="op">=</span> <span class="fl">4</span>, ncol <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>                      dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M1"</span>, <span class="st">"M2"</span>, <span class="st">"M3"</span>, <span class="st">"M4"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                      <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ind1"</span>, <span class="st">"Ind2"</span>, <span class="st">"Ind3"</span>, <span class="st">"Ind4"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run trimming function (with user input required for LOD selection)</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">trim_LG</span><span class="op">(</span>chromosome <span class="op">=</span> <span class="st">"CHR1"</span>,</span></span>
<span class="r-in"><span>                  map <span class="op">=</span> <span class="va">map_data</span>,</span></span>
<span class="r-in"><span>                  geno <span class="op">=</span> <span class="va">geno_matrix</span>,</span></span>
<span class="r-in"><span>                  pop_type <span class="op">=</span> <span class="st">"F2"</span>,</span></span>
<span class="r-in"><span>                  drop_outliers <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                  n_cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Access trimmed genotype matrix</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">trimmed_genotype</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

