<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>geneticMapR • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="geneticMapR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>geneticMapR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/vignettes/geneticMapR.Rmd" class="external-link"><code>vignettes/geneticMapR.Rmd</code></a></small>
      <div class="d-none name"><code>geneticMapR.Rmd</code></div>
    </div>

    
    
<p><code>geneticMapR</code> is an R package designed to simplify the
construction of genetic maps and QTL analysis using molecular markers in
F<sub>2</sub> bi-parental populations.</p>
<p>Genetic map construction is often overlooked in the genetic mapping
literature and can be time-consuming. This package aims to make the
process reproducible and to provide tools for QTL visualization and
plant genetic research.</p>
<p>Functions in <code>geneticMapR</code> and key mapping steps will be
illustrated using a real dataset from <em>Beta vulgaris</em> (table
beet), an outcrossing species. The F<sub>2</sub> population was derived
from crosses between Conical × Round and Cylindrical × Round inbred
lines, representing distinct root shapes. The challenges posed by the
outcrossing nature of <em>Beta vulgaris</em> will help highlight
practical solutions for common problems in genetic mapping.</p>
<p>My early genetic maps were difficult to reproduce and prone to
mistakes. I hope <code>geneticMapR</code> supports your journey toward
reliable and reproducible genetic maps.</p>
<div class="float">
<img src="../reference/figures/flatxcyl.png" alt="Flat x Cylindrical table beet cross. Image credit. Dr. Maddy Oravec"><div class="figcaption">Flat x Cylindrical table beet cross. Image
credit. Dr. Maddy Oravec</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get devtools if needed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">#Install geneticMapR</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"vegaalfaro/geneticMapR"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="where-to-start">Where to start?<a class="anchor" aria-label="anchor" href="#where-to-start"></a>
</h2>
<p><code>geneticMapR</code> helps you construct genetic maps in
F<sub>2</sub> mapping populations. While it is primarily designed for
diploid species, some functions also support polyploids.</p>
<p>To get started, you’ll need a few essential files:</p>
<ul>
<li><p><strong>Genotype file</strong>: A matrix where individuals are
scored for their genotype across multiple loci in the genome.</p></li>
<li><p><strong>Phenotype file</strong>: One or more traits measured
across individuals, possibly across time or environments
(multi-environment trials), depending on your species.</p></li>
</ul>
<p><code>geneticMapR</code> provides example data to walk you through
the mapping process. We’ll start with an example <code>VCF</code>[<a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" class="external-link uri">https://samtools.github.io/hts-specs/VCFv4.2.pdf</a>] file
which is the typical output of next generation sequencing platforms like
genotyping by sequencing (<a href="https://en.wikipedia.org/wiki/Genotyping_by_sequencing" class="external-link">GBS</a>)</p>
</div>
<div class="section level2">
<h2 id="recommendations">Recommendations<a class="anchor" aria-label="anchor" href="#recommendations"></a>
</h2>
<p>When genotyping, it is critical to include DNA from the parental
lines and the F<sub>1</sub> individuals. Ideally, this should be from
the actual plants used to create the population, especially in
outcrossing species like carrot or beet.</p>
<p>If tissue from the original parents or F<sub>1</sub> is unavailable,
use samples from other plants of the same genotype. However, in
outcrossing species, even inbred lines may show unexpected diversity.
I’ve seen this firsthand in carrots.</p>
<p>Parental genotypes are essential for <em>recoding</em> or
<em>phasing</em> markers to reflect parental inheritance, a core step in
genetic mapping. They’re also helpful for resolving problematic markers
and understanding residual heterozygosity, which is common in many
populations.</p>
<p>Having the parents and F<sub>1</sub> genotyped will prove invaluable
for a genetic mapping project.</p>
</div>
<div class="section level2">
<h2 id="using-your-own-data">Using Your Own Data<a class="anchor" aria-label="anchor" href="#using-your-own-data"></a>
</h2>
<p>You can use your own genotype matrix (from a VCF or GBS pipeline)
with <code>geneticMapR</code>. Functions like
<code><a href="../reference/convert_to_dosage.html">convert_to_dosage()</a></code> and
<code><a href="../reference/convert_to_dosage_flex.html">convert_to_dosage_flex()</a></code> help format raw genotype calls
into phased numeric values compatible with the package.</p>
<p>For unfiltered VCF-derived matrices, make sure to:</p>
<ul>
<li>Filter for bi-allelic SNPs.</li>
<li>Remove non-polymorphic markers.</li>
<li>Keep only SNPs with sufficient depth and genotype quality scores, it
varies for each project or population.</li>
</ul>
<p>If you need help in this step, check the
<code>bcftools_tutorial</code> in my github <a href="https://github.com/vegaalfaro/bcftoolsR" class="external-link">vegaalfaro/bcftoolsR</a>.
Still under development.</p>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
