<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Binning SNP Markers • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Binning SNP Markers">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../articles/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Binning SNP Markers</h1>
                        <h4 data-toc-skip class="author">Andrey Vega
Alfaro</h4>
            
            <h4 data-toc-skip class="date">2025-05-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/vignettes/LD_binning.Rmd" class="external-link"><code>vignettes/LD_binning.Rmd</code></a></small>
      <div class="d-none name"><code>LD_binning.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to bin SNP markers based on linkage
disequilibrium (LD) and detect genetic duplicates in a linkage map
F<sub>2</sub> table beet population using the <code>geneticMapR</code>
package and its dependencies. Specially useful is <a href="https://github.com/jendelman/MapRtools" class="external-link">MapRtools</a>.</p>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Helper function to install from CRAN if not available</span></span>
<span><span class="va">load_or_install_cran</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">pkg</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pkg</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Helper function to install from GitHub if not available</span></span>
<span><span class="va">load_or_install_github</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pkg</span>, <span class="va">repo</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">pkg</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="va">repo</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pkg</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Ensure devtools is available for GitHub installs</span></span>
<span><span class="fu">load_or_install_cran</span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install/load GitHub packages</span></span>
<span><span class="fu">load_or_install_github</span><span class="op">(</span><span class="st">"MapRtools"</span>, <span class="st">"jendelman/MapRtools"</span><span class="op">)</span></span>
<span><span class="fu">load_or_install_github</span><span class="op">(</span><span class="st">"geneticMapR"</span>, <span class="st">"vegaalfaro/geneticMapR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install/load CRAN packages</span></span>
<span><span class="fu">load_or_install_cran</span><span class="op">(</span><span class="st">"ggdendro"</span><span class="op">)</span></span>
<span><span class="fu">load_or_install_cran</span><span class="op">(</span><span class="st">"stringr"</span><span class="op">)</span></span>
<span><span class="fu">load_or_install_cran</span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>We begin by loading a preprocessed genotype dataset from our data
repository <a href="https://github.com/vegaalfaro/geneticMapRFiles" class="external-link">geneticMapRFiles</a></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno_matrices_url</span> <span class="op">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/vegaalfaro/geneticMapRFiles/main/R_data/filtered_geno_matrices_1629.RData"</span></span>
<span></span>
<span><span class="co"># Download file</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"local_copy.filtered_geno_matrices_1629.RData"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">geno_matrices_url</span>, </span>
<span>                destfile <span class="op">=</span> <span class="st">"local_copy.filtered_geno_matrices_1629.RData"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"local_copy.filtered_geno_matrices_1629.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="identify-genetic-duplicates">Identify Genetic Duplicates<a class="anchor" aria-label="anchor" href="#identify-genetic-duplicates"></a>
</h2>
<p>It’s often useful to include known duplicates in your dataset to
assess GBS quality. Generally it’s good to have a system that allows you
to figure out if you inadvertently included a genetic duplicate. In this
data, I included a duplicate of the F1 individual. Occasionally, there
is human error and some genetic duplicates are included in the
experiment inadvertently.</p>
<p>For convenience in the code below, we search for and assign a name to
the F1 and parental individuals.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assign names</span></span>
<span><span class="va">P1</span> <span class="op">&lt;-</span> <span class="st">"P2550-Cylindra-P1-Theta-A9"</span></span>
<span><span class="va">P2</span> <span class="op">&lt;-</span> <span class="st">"P2493-Mono-P2-Theta-B9"</span></span>
<span><span class="va">F1s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"7001-F1-Beta-H9"</span>, <span class="st">"7002-F1-Gamma-F11"</span><span class="op">)</span> </span></code></pre></div>
<p><code><a href="../reference/rename_geno_matrix.html">rename_geno_matrix()</a></code> standardizes the column names of a
genotype matrix by renaming parental genotypes (P1, P2), F1 individuals
(F1.1, F1.2, …), and F<sub>2</sub> individuals (F2.1, F2.2, …). This
shortens the names as they are quite large in the real data example and
they get in the way of visualization. We’ll change the names temporarily
for visualization purposes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># First make copy of geno</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="va">het_phased_geno_1629_filt</span></span>
<span><span class="co"># Rename</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rename_geno_matrix.html">rename_geno_matrix</a></span><span class="op">(</span><span class="va">geno</span>, <span class="va">P1</span>, <span class="va">P2</span>, <span class="va">F1s</span><span class="op">)</span></span></code></pre></div>
<p>We will use a dendrogram to determine genetic duplicates. The closer
two samples are to the lowest branches of the dedrogram, the more
related they are more likely to be duplicates. In this case, we can see
the two identical F1s (on the far right hand side) are on the
second-lowest branches. In addition, F24 and F25 appear to be even more
more closely related and are likely duplicates included in the
experiment because of human error. In most projects I have worked on,
surprises like this are not uncommon.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate distance matrix using  genotype matrix</span></span>
<span><span class="va">distance_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">geno</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform hierarchical clustering</span></span>
<span><span class="va">hclust_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="va">distance_matrix</span>, method <span class="op">=</span> <span class="st">"single"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert hclust object to dendrogram data for better visualization</span></span>
<span><span class="va">dendro_data</span> <span class="op">&lt;-</span> <span class="fu">ggdendro</span><span class="fu">::</span><span class="fu"><a href="https://andrie.github.io/ggdendro/reference/dendro_data.html" class="external-link">dendro_data</a></span><span class="op">(</span><span class="va">hclust_result</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot dendrogram using ggplot2</span></span>
<span><span class="va">dendro</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_segment</span><span class="op">(</span>data <span class="op">=</span> <span class="va">dendro_data</span><span class="op">$</span><span class="va">segments</span>, </span>
<span>               <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span>, xend <span class="op">=</span> <span class="va">xend</span>, yend <span class="op">=</span> <span class="va">yend</span><span class="op">)</span>, </span>
<span>               color <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">dendro_data</span><span class="op">$</span><span class="va">labels</span>, </span>
<span>            <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span>, label <span class="op">=</span> <span class="va">label</span><span class="op">)</span>, </span>
<span>            hjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">90</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Hierarchical Clustering Dendrogram"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Samples"</span>, y <span class="op">=</span> <span class="st">"Distance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_test</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, axis.ticks.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dendro</span></span></code></pre></div>
<p><img src="LD_binning_files/figure-html/unnamed-chunk-5-1.png" alt="dendro plot" width="777.6" style="display: block; margin: auto;"></p>
<div class="section level3">
<h3 id="filter-parents-and-f1-from-the-dataset">Filter Parents and F1 from the Dataset<a class="anchor" aria-label="anchor" href="#filter-parents-and-f1-from-the-dataset"></a>
</h3>
<p>To avoid bias during LD binning, we remove the <code>parents</code>
and F<sub>1</sub> individuals. We’ll keep the two F<sub>2s</sub> which
seem to be genetic duplicates for now.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hom_phased_geno_1629</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/drop_parents.html">drop_parents</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span>, <span class="va">P1</span>, <span class="va">P2</span>, F1 <span class="op">=</span> <span class="va">F1s</span><span class="op">)</span></span>
<span><span class="va">het_phased_geno_1629_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/drop_parents.html">drop_parents</a></span><span class="op">(</span><span class="va">het_phased_geno_1629_filt</span>, <span class="va">P1</span>, <span class="va">P2</span>, F1 <span class="op">=</span> <span class="va">F1s</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="bin-homozygous-markers">Bin Homozygous Markers<a class="anchor" aria-label="anchor" href="#bin-homozygous-markers"></a>
</h2>
<p>We begin by cleaning the homozygous matrix and removing
non-informative markers.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify monomorphic markers</span></span>
<span><span class="va">mono_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co"># Identify zero variance markers</span></span>
<span><span class="va">hom_phased_geno_1629</span> <span class="op">&lt;-</span> <span class="va">hom_phased_geno_1629</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span>, <span class="fl">1</span>, <span class="va">var</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="op">]</span></span>
<span><span class="co"># Filter them out from matrix</span></span>
<span><span class="va">hom_phased_geno_1629</span> <span class="op">&lt;-</span> <span class="va">hom_phased_geno_1629</span><span class="op">[</span><span class="op">!</span><span class="va">mono_markers</span>, <span class="op">]</span></span>
<span><span class="co"># If genotype matrix contains missing genotypes, remove them</span></span>
<span><span class="va">hom_phased_geno_1629</span> <span class="op">&lt;-</span> <span class="va">hom_phased_geno_1629</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># bin markers using MapRtools::LDbin</span></span>
<span><span class="va">LDbin_hom</span> <span class="op">&lt;-</span> <span class="fu">LDbin</span><span class="op">(</span><span class="va">hom_phased_geno_1629</span>, r2.thresh <span class="op">=</span> <span class="fl">0.99</span><span class="op">)</span></span>
<span><span class="co"># Extract binned genotype matrix</span></span>
<span><span class="va">geno.hom.bin</span> <span class="op">&lt;-</span> <span class="va">LDbin_hom</span><span class="op">$</span><span class="va">geno</span></span>
<span></span>
<span><span class="co"># Check dimensions of binned genotype</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">geno.hom.bin</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 998 100</span></span></code></pre></div>
<div class="section level3">
<h3 id="visualize-binned-homozygous-markers">Visualize Binned Homozygous Markers<a class="anchor" aria-label="anchor" href="#visualize-binned-homozygous-markers"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_map.html">extract_map</a></span><span class="op">(</span><span class="va">geno.hom.bin</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_coverage.html">plot_coverage</a></span><span class="op">(</span>map <span class="op">=</span> <span class="va">map_hom</span>, customize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Binned Homozygous Markers"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LD_binning_files/figure-html/unnamed-chunk-8-1.png" alt="coverageplot" width="588.48" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="bin-heterozygous-markers">Bin Heterozygous Markers<a class="anchor" aria-label="anchor" href="#bin-heterozygous-markers"></a>
</h2>
<p>We now repeat the binning process for heterozygous markers.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LDbin_het</span> <span class="op">&lt;-</span> <span class="fu">LDbin</span><span class="op">(</span><span class="va">het_phased_geno_1629_filt</span>, r2.thresh <span class="op">=</span> <span class="fl">0.99</span><span class="op">)</span></span>
<span><span class="va">geno.het.bin</span> <span class="op">&lt;-</span> <span class="va">LDbin_het</span><span class="op">$</span><span class="va">geno</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">geno.het.bin</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2480  100</span></span></code></pre></div>
<div class="section level3">
<h3 id="visualize-binned-heterozygous-markers">Visualize Binned Heterozygous Markers<a class="anchor" aria-label="anchor" href="#visualize-binned-heterozygous-markers"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_map.html">extract_map</a></span><span class="op">(</span><span class="va">geno.het.bin</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_coverage.html">plot_coverage</a></span><span class="op">(</span>map <span class="op">=</span> <span class="va">map_het</span>, customize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Binned Heterozygous Markers"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LD_binning_files/figure-html/unnamed-chunk-10-1.png" alt="coverageplot" width="588.48" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="mrker-filtering-tradeoff">Mrker Filtering Tradeoff<a class="anchor" aria-label="anchor" href="#mrker-filtering-tradeoff"></a>
</h2>
<p>This real-data example shows once more how filtering for homozygous
vs. heterozygous markers affects genome coverage. The balance between
marker type and map resolution.</p>
</div>
<div class="section level2">
<h2 id="save-results">Save Results<a class="anchor" aria-label="anchor" href="#save-results"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">geno.het.bin</span>, </span>
<span>     <span class="va">geno.hom.bin</span>,  </span>
<span>     <span class="va">dendro</span>,</span>
<span>     file <span class="op">=</span> <span class="st">"processed_data/R_data/binned_geno_1629.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This vignette illustrates how to identify duplicate individuals in
your dataset and perform LD-based marker binning using tools available
in the geneticMapR workflow.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.5.2          stringr_1.5.1          ggdendro_0.2.0        </span></span>
<span><span class="co">#&gt; [4] geneticMapR_0.0.0.9000 MapRtools_0.36         devtools_2.4.5        </span></span>
<span><span class="co">#&gt; [7] usethis_3.1.0         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1     Rmpfr_1.1-0          dplyr_1.1.4         </span></span>
<span><span class="co">#&gt;  [4] farver_2.1.2         fastmap_1.2.0        TSP_1.2-4           </span></span>
<span><span class="co">#&gt;  [7] promises_1.3.2       digest_0.6.37        mime_0.13           </span></span>
<span><span class="co">#&gt; [10] lifecycle_1.0.4      ellipsis_0.3.2       magrittr_2.0.3      </span></span>
<span><span class="co">#&gt; [13] compiler_4.5.0       rlang_1.1.6          sass_0.4.10         </span></span>
<span><span class="co">#&gt; [16] tools_4.5.0          yaml_2.3.10          knitr_1.50          </span></span>
<span><span class="co">#&gt; [19] ggsignif_0.6.4       labeling_0.4.3       htmlwidgets_1.6.4   </span></span>
<span><span class="co">#&gt; [22] bit_4.6.0            pkgbuild_1.4.8       RColorBrewer_1.1-3  </span></span>
<span><span class="co">#&gt; [25] pkgload_1.4.0        abind_1.4-8          miniUI_0.1.2        </span></span>
<span><span class="co">#&gt; [28] registry_0.5-1       ca_0.71.1            withr_3.0.2         </span></span>
<span><span class="co">#&gt; [31] purrr_1.0.4          desc_1.4.3           grid_4.5.0          </span></span>
<span><span class="co">#&gt; [34] urlchecker_1.0.1     ggpubr_0.6.0         profvis_0.4.0       </span></span>
<span><span class="co">#&gt; [37] xtable_1.8-4         MASS_7.3-65          scales_1.4.0        </span></span>
<span><span class="co">#&gt; [40] iterators_1.0.14     cli_3.6.5            rmarkdown_2.29      </span></span>
<span><span class="co">#&gt; [43] ragg_1.4.0           generics_0.1.4       remotes_2.5.0       </span></span>
<span><span class="co">#&gt; [46] CVXR_1.0-15          sessioninfo_1.2.3    cachem_1.1.0        </span></span>
<span><span class="co">#&gt; [49] splines_4.5.0        parallel_4.5.0       scam_1.2-18         </span></span>
<span><span class="co">#&gt; [52] vctrs_0.6.5          Matrix_1.7-3         jsonlite_2.0.0      </span></span>
<span><span class="co">#&gt; [55] HMM_1.0.2            carData_3.0-5        car_3.1-3           </span></span>
<span><span class="co">#&gt; [58] seriation_1.5.7      bit64_4.6.0-1        rstatix_0.7.2       </span></span>
<span><span class="co">#&gt; [61] Formula_1.2-5        systemfonts_1.2.3    ggdist_3.3.3        </span></span>
<span><span class="co">#&gt; [64] foreach_1.5.2        jquerylib_0.1.4      tidyr_1.3.1         </span></span>
<span><span class="co">#&gt; [67] glue_1.8.0           pkgdown_2.1.3        codetools_0.2-20    </span></span>
<span><span class="co">#&gt; [70] distributional_0.5.0 stringi_1.8.7        gtable_0.3.6        </span></span>
<span><span class="co">#&gt; [73] later_1.4.2          gmp_0.7-5            tibble_3.2.1        </span></span>
<span><span class="co">#&gt; [76] pillar_1.10.2        splines2_0.5.4       htmltools_0.5.8.1   </span></span>
<span><span class="co">#&gt; [79] R6_2.6.1             textshaping_1.0.1    evaluate_1.0.3      </span></span>
<span><span class="co">#&gt; [82] shiny_1.10.0         lattice_0.22-6       backports_1.5.0     </span></span>
<span><span class="co">#&gt; [85] memoise_2.0.1        broom_1.0.8          httpuv_1.6.16       </span></span>
<span><span class="co">#&gt; [88] bslib_0.9.0          Rcpp_1.0.14          nlme_3.1-168        </span></span>
<span><span class="co">#&gt; [91] mgcv_1.9-1           xfun_0.52            fs_1.6.6            </span></span>
<span><span class="co">#&gt; [94] pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
