<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Linkage Groups and Marker Order • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Linkage Groups and Marker Order">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Haplotype_and_Map.html">Genetic Map</a></li>
    <li><a class="dropdown-item" href="../articles/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../articles/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Linkage Groups and Marker Order</h1>
                        <h4 data-toc-skip class="author">Andrey Vega
Alfaro</h4>
            
            <h4 data-toc-skip class="date">2025-05-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/vignettes/Linkage_Groups.Rmd" class="external-link"><code>vignettes/Linkage_Groups.Rmd</code></a></small>
      <div class="d-none name"><code>Linkage_Groups.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p>This vignette provides an overview on how to estimate linkage groups,
and order markers using functions in the <code>geneticMapR</code> and
<code>MapRtools</code> packages. We continue to use genotype data from a
biparental F<sub>2</sub> table beet population.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>The package includes helper functions to check and install from CRAN
or GitHub. Feel free to use them as shown below if you need to install
packages needed for following along this article.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure devtools is available for GitHub installs</span></span>
<span><span class="fu"><a href="../reference/load_or_install_cran.html">load_or_install_cran</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install/load GitHub packages</span></span>
<span><span class="fu"><a href="../reference/load_or_install_github.html">load_or_install_github</a></span><span class="op">(</span><span class="st">"MapRtools"</span>, <span class="st">"jendelman/MapRtools"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_or_install_github.html">load_or_install_github</a></span><span class="op">(</span><span class="st">"geneticMapR"</span>, <span class="st">"vegaalfaro/geneticMapR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install/load CRAN packages</span></span>
<span><span class="fu"><a href="../reference/load_or_install_cran.html">load_or_install_cran</a></span><span class="op">(</span><span class="st">"ggplot2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_or_install_cran.html">load_or_install_cran</a></span><span class="op">(</span><span class="st">"tidyr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_or_install_cran.html">load_or_install_cran</a></span><span class="op">(</span><span class="st">"tidyverse"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_or_install_cran.html">load_or_install_cran</a></span><span class="op">(</span><span class="st">"parallel"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegaalfaro/geneticMapR" class="external-link">geneticMapR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MapRtools</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://andrie.github.io/ggdendro/" class="external-link">ggdendro</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data">1. Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>To avoid long runtimes, precomputed genotype data and LOD matrices
are loaded from <code>geneticMapRFiles</code>. The two genotype files
loaded are <code>geno.het.bin</code> and <code>geno.hom.bin</code>.</p>
<ul>
<li><p><code>geno.het.bin</code> is a genotype matrix of 2480 marker
bins by 100 individuals. Includes heterozygous and homozygous markers.
Rows are markers and rows are named (e.g.,CHR7_192222). Columns are
individuals and columns are named (e.g., 2002-F2-Beta-B10).</p></li>
<li><p><code>geno.hom.bin</code> is genotype matrix of 998 marker bins
by 100 individuals. Includes homozygous markers only. Rows are markers
with rownames (e.g.,CHR7_192222) and individuals are columns with column
names (e.g., 2002-F2-Beta-B10)..</p></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Binned Genotype</span></span>
<span><span class="va">url1</span> <span class="op">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/vegaalfaro/geneticMapRFiles/main/R_data/binned_geno_1629.RData"</span></span>
<span></span>
<span><span class="co"># Download file</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"binned_geno_1629.RData"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url1</span>, destfile <span class="op">=</span> <span class="st">"local_copy.binned_geno_1629.RData"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Load</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"local_copy.binned_geno_1629.RData"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="lod-matrices">LOD matrices<a class="anchor" aria-label="anchor" href="#lod-matrices"></a>
</h3>
<p>Precomputed LOD matrices will be loaded because they take a while to
run. The uploaded LOD matrices are:</p>
<ul>
<li><p><code>LODmat.het</code> Large matrix containing LOD scores for
each marker pair. Includes Heterozygous markers and Homozygous
markers.</p></li>
<li><p><code>LODmat.hom</code> Large matrix containing LOD scores for
each marker pair. Includes Homozygous markers only.</p></li>
</ul>
<p>(<strong>For info on marker types</strong>See marker type analysis
from the article “Recode Markers” under the Articles tab)</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># LOD matrices</span></span>
<span><span class="va">url2</span> <span class="op">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/vegaalfaro/geneticMapRFiles/main/R_data/LODmats_1629-2025-03-12.RData"</span></span>
<span></span>
<span><span class="co"># Download file</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"binned_geno_1629.RData"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url2</span>, destfile <span class="op">=</span> <span class="st">"LODmats_1629-2025-03-12.RData"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Load</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"LODmats_1629-2025-03-12.RData"</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="your-own-lod-matrix">Your own LOD matrix<a class="anchor" aria-label="anchor" href="#your-own-lod-matrix"></a>
</h4>
<p>If you need to run your own LOD matrices, below I show an example on
how to do so using <code>MapRtools</code> and the MLEL function. Make
sure to check how many cores you have available in your machine. The
example is for an F<sub>2</sub> population from two (inbred) parents.
However, MLEL supports several population types.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Geno matrix with only homozygous markers</span></span>
<span><span class="va">LODmat.hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MapRtools/man/MLEL.html" class="external-link">MLEL</a></span><span class="op">(</span>geno <span class="op">=</span> <span class="va">geno.hom.bin</span>, pop.type <span class="op">=</span> <span class="st">"F2"</span>, LOD<span class="op">=</span><span class="cn">TRUE</span>, n.core <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Geno matrix with homozygous and heterozygous markers</span></span>
<span><span class="va">LODmat.het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MapRtools/man/MLEL.html" class="external-link">MLEL</a></span><span class="op">(</span>geno <span class="op">=</span> <span class="va">geno.het.bin</span>, pop.type <span class="op">=</span> <span class="st">"F2"</span>, LOD<span class="op">=</span><span class="cn">TRUE</span>, n.core <span class="op">=</span> <span class="fl">7</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="linkage-groups">2. Linkage Groups<a class="anchor" aria-label="anchor" href="#linkage-groups"></a>
</h2>
<p>The objective in determining linkage groups is to identify the
smallest LOD (logarithm of odds) threshold at which all chromosomes are
correctly separated into their own linkage groups. Linkage groups are
clusters of genetic markers that are inherited together and correspond
to physical chromosomes. The goal is to identify these groups by
assessing how often markers are co-inherited across progeny of, in this
case, an F<sub>2</sub> mapping population. So:</p>
<p>Markers that are:</p>
<p>Close together on the same chromosome tend to be inherited together
(low recombination → high linkage).</p>
<p>Far apart or on different chromosomes are inherited independently
(high recombination → low or no linkage).</p>
<div class="section level3">
<h3 id="lod-threshold-tradeoffs">LOD threshold tradeoffs<a class="anchor" aria-label="anchor" href="#lod-threshold-tradeoffs"></a>
</h3>
<p>The LOD threshold determines how stringently you connect markers into
groups:</p>
<ul>
<li><p>Too low: Spurious linkage → falsely merges different
chromosomes.</p></li>
<li><p>Too high: Real linkage missed → splits chromosomes, especially
if:</p></li>
</ul>
<p>There are long arms with low marker density (low resolution), or
recombination hotspots create perceived gaps.</p>
<p>So, for beet (2n=18; 9 chromosomes), we’re trying to tune the LOD
threshold so that exactly 9 linkage groups emerge, matching the known
number of chromosomes.</p>
<p>This is often done empirically:</p>
<ul>
<li><p>Estimate pairwise LOD scores between all markers.</p></li>
<li><p>Build a graph or a figure to visualize linked markers.</p></li>
<li><p>Adjust the threshold to get the correct number of groups (for
example, by counting connected components in a graph).</p></li>
</ul>
<p>We will get to this shortly, but before let me show you some theory
behind it.</p>
</div>
<div class="section level3">
<h3 id="some-math-behind-lod-scores">Some math behind LOD scores<a class="anchor" aria-label="anchor" href="#some-math-behind-lod-scores"></a>
</h3>
<p>The LOD score compares two hypotheses:</p>
<ul>
<li>H1: The two markers are linked at a certain recombination frequency
(θ &lt; 0.5).</li>
<li>H0: The two markers are not linked (θ = 0.5, i.e., independent
assortment).</li>
</ul>
<p>The formula is:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>L</mi><mi>O</mi><mi>D</mi><mo>=</mo><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>10</mn></msub><mrow><mo stretchy="true" form="prefix">(</mo><mfrac><mrow><mi>P</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>d</mi><mi>a</mi><mi>t</mi><mi>a</mi><mo>∣</mo><mi>l</mi><mi>i</mi><mi>n</mi><mi>k</mi><mi>e</mi><mi>d</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><mi>P</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>d</mi><mi>a</mi><mi>t</mi><mi>a</mi><mo>∣</mo><mi>u</mi><mi>n</mi><mi>l</mi><mi>i</mi><mi>n</mi><mi>k</mi><mi>e</mi><mi>d</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">LOD=log_{10}\left(\frac{P(data∣linked)}{P(data∣unlinked)}\right) </annotation></semantics></math></p>
<p>A higher LOD means stronger evidence that the markers belong to the
same group. Usually a treshold is set to define linkage groups.</p>
<p>To visualize linkage group separation, you can use the
<code>MapRtools</code> package to generate a candy stripe bar plot. This
plot helps assess whether chromosomes have been cleanly split.</p>
</div>
<div class="section level3">
<h3 id="linkage-group-separation">Linkage group separation<a class="anchor" aria-label="anchor" href="#linkage-group-separation"></a>
</h3>
<p>Let’s use <code>MapRtools</code> <code>LG</code> function to estimate
linkage group separation. You can toggle with the threshold. You can try
different thresholds for example start 2 to 20 and move up iteratively.
By counting the stripes we can figure out which threshold is
appropriate. The black line on top represent misplaced markers or
markers in Linkage Groups outside the largest first 9.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MapRtools/man/LG.html" class="external-link">LG</a></span><span class="op">(</span><span class="va">LODmat.hom</span>, thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">30</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Homozygous markers only"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Linkage_Groups_files/figure-html/unnamed-chunk-7-1.png" width="624"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MapRtools/man/LG.html" class="external-link">LG</a></span><span class="op">(</span><span class="va">LODmat.het</span>, thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">30</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Heterozygous + Homozygous SNPs"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Linkage_Groups_files/figure-html/unnamed-chunk-7-2.png" width="624"></p>
<p>For <strong>homozygous markers</strong>, nine linkage groups are
first formed at a LOD score of 12. This configuration remains stable up
to a LOD of 18, which is support for the grouping structure and suggests
reliable marker clustering.</p>
<p>When <strong>heterozygous markers</strong> are included alongside
homozygous markers, the pattern shifts slightly. In this combined
dataset, nine linkage groups are formed at a higher LOD score of 19.
However, the grouping is less clear-cut than in the homozygous-only
scenario.</p>
</div>
</div>
<div class="section level2">
<h2 id="linkage-group-separation-1">3. Linkage Group Separation<a class="anchor" aria-label="anchor" href="#linkage-group-separation-1"></a>
</h2>
<p>We’re going to demonstrate the next steps with the set of homozygous
markers for convenience. In reality you may have a combination of
homozygous and heterozygous markers. It all depends on your marker set
and which steps you took to filter them. Other articles of this package
show ways to filter markers. Make sure to check them out.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_map.html">extract_map</a></span><span class="op">(</span><span class="va">geno.hom.bin</span>, markers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># Extracts physical map</span></span>
<span><span class="va">LGhom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MapRtools/man/LG.html" class="external-link">LG</a></span><span class="op">(</span><span class="va">LODmat.hom</span>, thresh <span class="op">=</span> <span class="fl">12</span><span class="op">)</span> <span class="co"># Returns assiged linkage group</span></span>
<span><span class="va">LGhom</span><span class="op">$</span><span class="va">chrom</span> <span class="op">&lt;-</span> <span class="va">map</span><span class="op">$</span><span class="va">chrom</span> </span>
<span><span class="va">LGhom</span> <span class="op">&lt;-</span> <span class="va">LGhom</span><span class="op">[</span><span class="va">LGhom</span><span class="op">$</span><span class="va">LG</span> <span class="op">&lt;=</span> <span class="fl">9</span>, <span class="op">]</span> <span class="co"># Keeps largest 9 linkage groups</span></span>
<span><span class="va">map</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">LGhom</span>, <span class="va">map</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"marker"</span>, <span class="st">"chrom"</span><span class="op">)</span><span class="op">)</span> <span class="co"># Merge to connect CHR to LGs</span></span>
<span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">map</span><span class="op">$</span><span class="va">chrom</span>, <span class="va">map</span><span class="op">$</span><span class="va">LG</span><span class="op">)</span> <span class="co"># Make a table of CHR and LGs</span></span>
<span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="va">tab</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/maxCol.html" class="external-link">max.col</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span>, ties.method <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="co"># Order table</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span> <span class="co"># Print table</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">6</th>
<th align="right">7</th>
<th align="right">8</th>
<th align="right">5</th>
<th align="right">4</th>
<th align="right">3</th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">CHR1</td>
<td align="right">102</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CHR2</td>
<td align="right">2</td>
<td align="right">82</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CHR3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">70</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CHR4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">109</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CHR5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">103</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CHR6</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">133</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CHR7</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">164</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CHR8</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">145</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">CHR9</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">61</td>
</tr>
</tbody>
</table>
<p>When the thresh argument of the <code><a href="https://rdrr.io/pkg/MapRtools/man/LG.html" class="external-link">MapRtools::LG()</a></code> function
is provided as a single numeric value, the function returns a data frame
with the assigned linkage group for each marker.</p>
<p>Small groups with only a few markers are likely artifacts of sparse
coverage or noise. For example chromosome 1 has most markers on linkage
group 6, however, linkage group 6 also encompasses a few markers from
chromosomes 2, 7 and 9. To focus on biologically meaningful results, we
subset the largest nine linkage groups, which correspond to the expected
number of chromosomes in beet.</p>
</div>
<div class="section level2">
<h2 id="linkage-trimming-with-trim_lg">4. Linkage Trimming with <code>trim_LG()</code><a class="anchor" aria-label="anchor" href="#linkage-trimming-with-trim_lg"></a>
</h2>
<p>When working with high-density genotype datasets, it’s important to
identify and remove markers that do not belong to the same linkage group
(LG) as the majority. Take for example our example of LG 6 which should
contain only markers belonging to chromosome 1. This process, called
linkage trimming, helps improve the accuracy of genetic map
construction.</p>
<p>We developed the function <code><a href="../reference/trim_LG.html">trim_LG()</a></code> to streamline this
process. It builds on <code>MapRtools</code> and allows you to:</p>
<ul>
<li><p>Filter markers based on LOD score thresholds for LG
assignment</p></li>
<li><p>Visualize haplotype frequencies before and after
trimming</p></li>
<li><p>Interactively define LOD parameters, or automate them by
providing the values directly</p></li>
<li><p>Remove outlier markers based on haplotype frequencies</p></li>
</ul>
<p>If you’re working with several chromosomes, or populations, and
already have a reference genome that provides marker order, this
function helps you <em>trim away spurious markers</em> without the need
to re-order them. However, if no reference is available, we recommend
ordering the markers first (e.g., with order_markers() from
MapRtools).</p>
<p>The code chunk below will:</p>
<ul>
<li><p>Estimate LOD scores for markers in CHR1</p></li>
<li><p>Plot the initial haplotype frequency</p></li>
<li><p>Generate a plot across a LOD range (5 to 35, in steps of
1)</p></li>
<li><p>Apply a final threshold of 18 to assign markers to the primary
LG</p></li>
<li><p>Optionally remove outlier markers</p></li>
<li><p>Return a trimmed genotype matrix, filtered map, LOD parameters
used, and both plots</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># CHR1</span></span>
<span><span class="va">result_chr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/trim_LG.html">trim_LG</a></span><span class="op">(</span>chromosome <span class="op">=</span> <span class="st">"CHR1"</span>, </span>
<span>                                  map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                                  pop_type <span class="op">=</span> <span class="st">"F2"</span>,</span>
<span>                                  geno <span class="op">=</span> <span class="va">geno.hom.bin</span>, <span class="co"># geno from the ordering step</span></span>
<span>                       interactive <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                       drop_outliers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                       n_cores <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                       initial_LOD <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                       end_LOD <span class="op">=</span> <span class="fl">35</span>,</span>
<span>                       step_LOD <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                       selected_thresh <span class="op">=</span> <span class="fl">18</span></span>
<span>                       </span>
<span>                                  <span class="op">)</span></span>
<span><span class="co">#&gt; Processing chromosome: CHR1</span></span>
<span><span class="co">#&gt; Plotting initial haplotype frequency...</span></span></code></pre></div>
<p><img src="Linkage_Groups_files/figure-html/unnamed-chunk-9-1.png" width="624"></p>
<pre><code><span><span class="co">#&gt; Estimating LOD matrix using 3 cores...</span></span>
<span><span class="co">#&gt; Generating linkage group plot...</span></span></code></pre>
<p><img src="Linkage_Groups_files/figure-html/unnamed-chunk-9-2.png" width="624"></p>
<pre><code><span><span class="co">#&gt; Filtering markers at LOD threshold = 18</span></span>
<span><span class="co">#&gt; Removing outliers...</span></span></code></pre>
<p>This function is also interactive! if you you set the argument
<code>interactive = TRUE</code> then the function requests the input of
the user and you can type in the console the initial LOD, end LOD and
the step. It will show you a plot and then you can select a threshold
for linkage grouping.</p>
<p>The goal is to select a threshold that represents most of the markers
as one linkage group and pushes away spurious markers that do not
belong.</p>
</div>
<div class="section level2">
<h2 id="ordering-markers-with-order_and_plot">5. Ordering Markers with <code>order_and_plot()</code><a class="anchor" aria-label="anchor" href="#ordering-markers-with-order_and_plot"></a>
</h2>
<p>After trimming spurious markers from each chromosome, it’s important
to optimize their order based on recombination frequencies. The function
<code><a href="../reference/order_and_plot.html">order_and_plot()</a></code> helps streamline this step by combining
marker ordering and haplotype visualization into a single workflow.</p>
<p>This function:</p>
<ul>
<li><p>Estimates the recombination frequency (RF) matrix with
<code><a href="https://rdrr.io/pkg/MapRtools/man/MLEL.html" class="external-link">MapRtools::MLEL()</a></code></p></li>
<li><p>Performs multiple iterations of
<code><a href="https://rdrr.io/pkg/MapRtools/man/order_markers.html" class="external-link">MapRtools::order_markers()</a></code></p></li>
<li><p>Selects the best marker order based on the smallest Sum of
Adjusted Recombination Fractions (SARF)</p></li>
<li><p>Generates a faceted plot of all ordering attempts for visual
inspection</p></li>
<li><p>Displays a haplotype plot for a subset of individuals in the
ordered matrix</p></li>
</ul>
<p>It is designed to operate on trimmed genotype matrices, typically one
chromosome at a time.</p>
<div class="section level3">
<h3 id="example-ordering-markers-in-a-single-chromosome">Example: Ordering markers in a single chromosome<a class="anchor" aria-label="anchor" href="#example-ordering-markers-in-a-single-chromosome"></a>
</h3>
<p>This code chunk will:</p>
<ul>
<li><p>Attempt 6 marker orderings for CHR1</p></li>
<li><p>Plot the original vs. optimized order for each iteration</p></li>
<li><p>Select the best order (lowest SARF)</p></li>
</ul>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Chromosome 1</span></span>
<span><span class="va">chr1_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/order_and_plot.html">order_and_plot</a></span><span class="op">(</span><span class="va">result_chr1</span><span class="op">$</span><span class="va">trimmed_genotype</span>, </span>
<span>                             n.iter <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                             CHR <span class="op">=</span> <span class="st">"CHR1"</span></span>
<span>                             <span class="op">)</span></span></code></pre></div>
<p><img src="Linkage_Groups_files/figure-html/unnamed-chunk-10-1.png" width="624"></p>
<p>We can observe some correspondence between the original marker order
and the optimized order. Occasionally, the order may appear reversed.
These plots help identify potential inversions or other chromosomal
features. In this section, the focus is on demonstrating the process of
marker ordering. The biological interpretation depends on the crop’s
genome structure and the quality of the reference genome. Use your
intuition and understanding of the underlying biology to interpret the
patterns in the plots.</p>
<p><code><a href="../reference/order_and_plot.html">order_and_plot()</a></code> uses MapRtools
<code><a href="https://rdrr.io/pkg/MapRtools/man/order_markers.html" class="external-link">order_markers()</a></code> which is based on solving the traveling
salesperson problem.</p>
</div>
<div class="section level3">
<h3 id="efficient-multiple-chromosome-ordering">Efficient multiple chromosome ordering<a class="anchor" aria-label="anchor" href="#efficient-multiple-chromosome-ordering"></a>
</h3>
<p>If you’ve trimmed and stored genotype matrices for each chromosome in
a named list, you can apply order_and_plot() in a loop:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  CHR1 <span class="op">=</span> <span class="va">result_chr1</span><span class="op">$</span><span class="va">trimmed_genotype</span>,</span>
<span>  CHR2 <span class="op">=</span> <span class="va">result_chr2</span><span class="op">$</span><span class="va">trimmed_genotype</span>,</span>
<span>  <span class="co"># ...</span></span>
<span>  CHR9 <span class="op">=</span> <span class="va">result_chr9</span><span class="op">$</span><span class="va">trimmed_genotype</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">ordered_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">result_list</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">chr</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/order_and_plot.html">order_and_plot</a></span><span class="op">(</span><span class="va">result_list</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, n.iter <span class="op">=</span> <span class="fl">6</span>, CHR <span class="op">=</span> <span class="va">chr</span>, prop <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ordered_results</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">result_list</span><span class="op">)</span></span></code></pre></div>
<p>This approach allows efficient and reproducible marker ordering
across your entire genome.</p>
</div>
</div>
<div class="section level2">
<h2 id="save-progress">Save progress<a class="anchor" aria-label="anchor" href="#save-progress"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Save results as RDA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">ordered_results</span>, file <span class="op">=</span> <span class="st">"processed_data/R_data/ordered_markers_pop2_list-2025-03-12.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This vignette outlines a complete workflow for LOD estimation,
linkage group assignment, and marker ordering for an F2 population using
SNPs.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
