<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Phenotypic Data Part II • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Phenotypic Data Part II">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../vignettes/Recode.html">Recode Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno.html">Phenotype Part I</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno_part_2.html">Phenotype Part II</a></li>
    <li><a class="dropdown-item" href="../vignettes/qtl_analysis.html">Linkage Groups and Marker Order</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Phenotypic Data Part II</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/vignettes/pheno_part_2.Rmd" class="external-link"><code>vignettes/pheno_part_2.Rmd</code></a></small>
      <div class="d-none name"><code>pheno_part_2.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p>This vignette will deal with how to properly and in a reproducible
way work your phenotypic and genotypic data in a format compatible with
R/qtl, a widely used package in linkage mapping.</p>
<p>The main <strong>objective of this article</strong> is to demonstrate
the use of the <code>format_qtl_input</code> function. This function
formats the genotype, map, and phenotype files for QTL analysis in
R/qtl. It allows the user to specify whether the genotype data should be
converted from dosage to ABH format or used as is.</p>
<p>So we need 3 critical files:</p>
<ul>
<li>genotype file</li>
<li>map</li>
<li>phenotype file</li>
</ul>
<p>Let’s get those and it should be really easy then to use the function
to put them together in a format that R/qtl can understand.</p>
<p>The main advantage of using this function is that it automates the
integration of map, phenotype, and genotype data. If you need to redo
the phenotypic analysis later, having a script that allows you to
reproducibly update your files will make the process much easier.</p>
</div>
<div class="section level2">
<h2 id="libraries">Libraries<a class="anchor" aria-label="anchor" href="#libraries"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegaalfaro/geneticMapR" class="external-link">geneticMapR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mjskay.github.io/ggdist/" class="external-link">ggdist</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rlang.r-lib.org" class="external-link">rlang</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org" class="external-link">readxl</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<div class="section level3">
<h3 id="phenotype">Phenotype<a class="anchor" aria-label="anchor" href="#phenotype"></a>
</h3>
<p>We continue with the data of our previous script.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Phenotype</span></span>
<span><span class="va">url1</span> <span class="op">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/vegaalfaro/geneticMapRFiles/refs/heads/main/phenotype/phenotypes-binary.csv"</span></span>
<span></span>
<span><span class="co"># Download file</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"local_copy_phenotypes-binary.csv"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url1</span>, destfile <span class="op">=</span> <span class="st">"local_copy_phenotypes-binary.csv"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Read</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"local_copy_phenotypes-binary.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Most important variables</span></span>
<span><span class="va">vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">16</span>, <span class="fl">22</span>, <span class="fl">23</span>, <span class="fl">24</span>, <span class="fl">26</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">response_vars</span> <span class="op">&lt;-</span> <span class="va">vars</span> <span class="co">#make a copy</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Preview of the first 7 columns of the phenotype dataset"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Preview of the first 7 columns of the phenotype
dataset</caption>
<thead><tr class="header">
<th align="right">ID</th>
<th align="left">Row</th>
<th align="left">UID</th>
<th align="left">Gen</th>
<th align="left">Geno</th>
<th align="right">F1Num</th>
<th align="right">Scale</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1001</td>
<td align="left">1204</td>
<td align="left">1021-MAP-2024</td>
<td align="left">F2</td>
<td align="left">1592-ConicalxRound-F2</td>
<td align="right">1592</td>
<td align="right">955</td>
</tr>
<tr class="even">
<td align="right">1002</td>
<td align="left">1204</td>
<td align="left">1022-MAP-2024</td>
<td align="left">F2</td>
<td align="left">1592-ConicalxRound-F2</td>
<td align="right">1592</td>
<td align="right">955</td>
</tr>
<tr class="odd">
<td align="right">1003</td>
<td align="left">1204</td>
<td align="left">1023-MAP-2024</td>
<td align="left">F2</td>
<td align="left">1592-ConicalxRound-F2</td>
<td align="right">1592</td>
<td align="right">945</td>
</tr>
<tr class="even">
<td align="right">1004</td>
<td align="left">1204</td>
<td align="left">1024-MAP-2024</td>
<td align="left">F2</td>
<td align="left">1592-ConicalxRound-F2</td>
<td align="right">1592</td>
<td align="right">947</td>
</tr>
<tr class="odd">
<td align="right">1005</td>
<td align="left">1204</td>
<td align="left">1025-MAP-2024</td>
<td align="left">F2</td>
<td align="left">1592-ConicalxRound-F2</td>
<td align="right">1592</td>
<td align="right">955</td>
</tr>
<tr class="even">
<td align="right">1006</td>
<td align="left">1204</td>
<td align="left">1026-MAP-2024</td>
<td align="left">F2</td>
<td align="left">1592-ConicalxRound-F2</td>
<td align="right">1592</td>
<td align="right">955</td>
</tr>
</tbody>
</table>
<p>In addition, we will need some meta-data available from the same data
repository <code>geneticMapRFiles</code>. The file ID contains the
column <code>VCF_name</code> which will match the ID in the genotype
matrix. The phenotype and the genotype files share the same individuals
but the names are slightly different. This ID file, 1001-F2-Alpha-A1,
1002-F2-Alpha-B1, 1003-F2-Alpha-C1). They share parto of the name but we
need to make them match.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">url2</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/vegaalfaro/geneticMapRFiles/raw/main/IDs/ID.csv"</span></span>
<span></span>
<span><span class="co"># Download file</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"local_copy_ID.csv"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url2</span>, destfile <span class="op">=</span> <span class="st">"local_copy_ID.csv"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Read</span></span>
<span><span class="va">ID_Pedigrees</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"local_copy_ID.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make as character</span></span>
<span><span class="va">ID_Pedigrees</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">ID_Pedigrees</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ID_Pedigrees</span><span class="op">)</span></span>
<span><span class="co">#&gt;     ID Sample_Name Population Type         VCF_name</span></span>
<span><span class="co">#&gt; 1 1001     1001-F2       1592   F2 1001-F2-Alpha-A1</span></span>
<span><span class="co">#&gt; 2 1002     1002-F2       1592   F2 1002-F2-Alpha-B1</span></span>
<span><span class="co">#&gt; 3 1003     1003-F2       1592   F2 1003-F2-Alpha-C1</span></span>
<span><span class="co">#&gt; 4 1004     1004-F2       1592   F2 1004-F2-Alpha-D1</span></span>
<span><span class="co">#&gt; 5 1005     1005-F2       1592   F2 1005-F2-Alpha-E1</span></span>
<span><span class="co">#&gt; 6 1006     1006-F2       1592   F2 1006-F2-Alpha-F1</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ID_Pedigrees</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Preview of the ID Meta-data"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Preview of the ID Meta-data</caption>
<thead><tr class="header">
<th align="left">ID</th>
<th align="left">Sample_Name</th>
<th align="left">Population</th>
<th align="left">Type</th>
<th align="left">VCF_name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1001</td>
<td align="left">1001-F2</td>
<td align="left">1592</td>
<td align="left">F2</td>
<td align="left">1001-F2-Alpha-A1</td>
</tr>
<tr class="even">
<td align="left">1002</td>
<td align="left">1002-F2</td>
<td align="left">1592</td>
<td align="left">F2</td>
<td align="left">1002-F2-Alpha-B1</td>
</tr>
<tr class="odd">
<td align="left">1003</td>
<td align="left">1003-F2</td>
<td align="left">1592</td>
<td align="left">F2</td>
<td align="left">1003-F2-Alpha-C1</td>
</tr>
<tr class="even">
<td align="left">1004</td>
<td align="left">1004-F2</td>
<td align="left">1592</td>
<td align="left">F2</td>
<td align="left">1004-F2-Alpha-D1</td>
</tr>
<tr class="odd">
<td align="left">1005</td>
<td align="left">1005-F2</td>
<td align="left">1592</td>
<td align="left">F2</td>
<td align="left">1005-F2-Alpha-E1</td>
</tr>
<tr class="even">
<td align="left">1006</td>
<td align="left">1006-F2</td>
<td align="left">1592</td>
<td align="left">F2</td>
<td align="left">1006-F2-Alpha-F1</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="genotype-data">Genotype data<a class="anchor" aria-label="anchor" href="#genotype-data"></a>
</h3>
<p>We’ll load the ordered genotype matrix that we worked on in the
previous scripts and is avaialbe in the the GitHub repository <a href="https://github.com/vegaalfaro/geneticMapRFiles/tree/main?tab=readme-ov-file" class="external-link">geneticMapRFiles</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">url3</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/vegaalfaro/geneticMapRFiles/raw/main/R_data/genotype_matrices_hmm_pop2-2025-03-12.RData"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url3</span>, destfile <span class="op">=</span> <span class="st">"local_copy_geno.RData"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"local_copy_geno.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="maps">Maps<a class="anchor" aria-label="anchor" href="#maps"></a>
</h3>
<p>These code loads the maps that we worked on and saved in the previous
<a href="https://vegaalfaro.github.io/geneticMapR/articles/Haplotype_and_Map.html">script</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">url4</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/vegaalfaro/geneticMapRFiles/raw/main/R_data/genetic_maps_pop2-2025-06-06.RData"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url4</span>, destfile <span class="op">=</span> <span class="st">"local_copy_maps.RData"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"local_copy_maps.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="prepare-data-for-rqtl">Prepare data for R/QTL<a class="anchor" aria-label="anchor" href="#prepare-data-for-rqtl"></a>
</h2>
<div class="section level3">
<h3 id="phenotype-1">Phenotype<a class="anchor" aria-label="anchor" href="#phenotype-1"></a>
</h3>
<p>This section filters the dataset to include only F<sub>2</sub>
individuals, excluding parents and F<sub>1s</sub>. It also matches
phenotype IDs to their corresponding VCF IDs for integration with
genotypic data, and selects only the relevant columns required for
downstream analysis in R/qtl.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># We subset like we did in the previous script.</span></span>
<span><span class="va">Pop2</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span><span class="va">pheno</span><span class="op">$</span><span class="va">Geno</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1629-CylindricalxFlat-F2"</span>, </span>
<span>                                <span class="st">"Cylindra"</span>, </span>
<span>                                <span class="st">"F1-1629"</span>, </span>
<span>                                <span class="st">"Mono"</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Keep only F2 Individuals</span></span>
<span><span class="va">Pop2_progeny</span> <span class="op">&lt;-</span> <span class="va">Pop2</span><span class="op">[</span><span class="va">Pop2</span><span class="op">$</span><span class="va">Gen</span> <span class="op">==</span> <span class="st">"F2"</span>,<span class="op">]</span></span>
<span><span class="va">Pop2_progeny</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">Pop2_progeny</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span> <span class="co"># Declare ID as character</span></span>
<span></span>
<span><span class="co"># Add VCF ID (ID that matches the geno file)</span></span>
<span><span class="va">Pop2_progeny</span> <span class="op">&lt;-</span> <span class="va">Pop2_progeny</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">ID_Pedigrees</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ID</span>, <span class="va">VCF_name</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure "VCF_name" is included in the selection</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">vars</span>, <span class="st">"VCF_name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Select only the desired columns and rename VCF_name to ID</span></span>
<span><span class="va">Pop2_progeny</span> <span class="op">&lt;-</span> <span class="va">Pop2_progeny</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="co"># Remove any existing grouping (Annoying dplyr thing)</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>ID <span class="op">=</span> <span class="va">VCF_name</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Pop2_progeny</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Preview of the columns of the phenotypic data"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Preview of the columns of the phenotypic data</caption>
<colgroup>
<col width="9%">
<col width="9%">
<col width="8%">
<col width="8%">
<col width="18%">
<col width="12%">
<col width="13%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th align="right">biomass</th>
<th align="right">max_width</th>
<th align="right">width_50</th>
<th align="right">length</th>
<th align="right">length_width_ratio</th>
<th align="right">shoulder_top</th>
<th align="right">tip_angle_top</th>
<th align="left">ID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">7046.486</td>
<td align="right">89.0997</td>
<td align="right">79.57</td>
<td align="right">111.6102</td>
<td align="right">1.25</td>
<td align="right">128.2565</td>
<td align="right">41.445822</td>
<td align="left">2001-F2-Beta-A10</td>
</tr>
<tr class="even">
<td align="right">5883.917</td>
<td align="right">72.3297</td>
<td align="right">72.12</td>
<td align="right">105.1587</td>
<td align="right">1.45</td>
<td align="right">108.9967</td>
<td align="right">49.626571</td>
<td align="left">2002-F2-Beta-B10</td>
</tr>
<tr class="odd">
<td align="right">8650.555</td>
<td align="right">88.8501</td>
<td align="right">60.36</td>
<td align="right">149.8485</td>
<td align="right">1.69</td>
<td align="right">124.2265</td>
<td align="right">57.371546</td>
<td align="left">2003-F2-Beta-C10</td>
</tr>
<tr class="even">
<td align="right">8809.267</td>
<td align="right">86.2728</td>
<td align="right">85.23</td>
<td align="right">135.5865</td>
<td align="right">1.57</td>
<td align="right">197.8657</td>
<td align="right">44.441035</td>
<td align="left">2004-F2-Beta-D10</td>
</tr>
<tr class="odd">
<td align="right">10023.938</td>
<td align="right">110.4552</td>
<td align="right">110.35</td>
<td align="right">119.0250</td>
<td align="right">1.08</td>
<td align="right">341.3724</td>
<td align="right">62.982283</td>
<td align="left">2005-F2-Beta-E10</td>
</tr>
<tr class="even">
<td align="right">10861.375</td>
<td align="right">92.2389</td>
<td align="right">86.41</td>
<td align="right">170.6049</td>
<td align="right">1.85</td>
<td align="right">256.2808</td>
<td align="right">5.314546</td>
<td align="left">2006-F2-Beta-F10</td>
</tr>
</tbody>
</table>
<p>Please notice that the phenotype file <code>Pop2_progeny</code> of
the which we got a glimpse, has the ID at the end. The ID column is not
really needed for R/qtl, in fact it detects it as another phenotype
trait. So we have the phenotype ready. Let’s move on to the map
file.</p>
</div>
<div class="section level3">
<h3 id="map">Map<a class="anchor" aria-label="anchor" href="#map"></a>
</h3>
<p>We will use the map that uses the reference genome order. The
function <code>format_qtl_input</code> requires the columns marker,
chrom and position. So we’ll change our file slightly to accomodate the
requirements.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Map</span></span>
<span><span class="va">map2</span> <span class="op">&lt;-</span> <span class="va">MAP_Ref_Genome_Order</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>marker <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">chrom</span>, <span class="st">"_"</span>, <span class="va">position_Mb</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">marker</span>, <span class="va">chrom</span>, position <span class="op">=</span> <span class="va">position_cM</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="format-data-with-format_qtl_input">Format data with <code>format_qtl_input</code><a class="anchor" aria-label="anchor" href="#format-data-with-format_qtl_input"></a>
</h2>
<p>The code below uses the <code>format_qtl_input</code> to convert the
phenotype, genetic map and genotype matrix in a format that can be used
to load the data in R/qtl.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example usage</span></span>
<span> <span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_qtl_input.html">format_qtl_input</a></span><span class="op">(</span><span class="va">geno_genome_order</span>, <span class="va">map2</span>, <span class="va">Pop2_progeny</span>, numeric <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in format_qtl_input(geno_genome_order, map2, Pop2_progeny, numeric =</span></span>
<span><span class="co">#&gt; TRUE): Warning: Some IDs in pheno do not match column names in geno.</span></span>
<span></span>
<span><span class="co"># Save results as csv </span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">result</span>, file <span class="op">=</span> <span class="st">"../data/rqtl.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Let’s look at the data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Preview of csv file ready for R/qtl"</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>Preview of csv file ready for R/qtl</caption>
<colgroup>
<col width="5%">
<col width="5%">
<col width="4%">
<col width="4%">
<col width="10%">
<col width="7%">
<col width="7%">
<col width="9%">
<col width="5%">
<col width="6%">
<col width="6%">
<col width="6%">
<col width="6%">
<col width="6%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">biomass</th>
<th align="left">max_width</th>
<th align="left">width_50</th>
<th align="left">length</th>
<th align="left">length_width_ratio</th>
<th align="left">shoulder_top</th>
<th align="left">tip_angle_top</th>
<th align="left">ID</th>
<th align="left">CHR1_98645</th>
<th align="left">CHR1_235900</th>
<th align="left">CHR1_312228</th>
<th align="left">CHR1_473415</th>
<th align="left">CHR1_635793</th>
<th align="left">CHR1_672859</th>
<th align="left">CHR1_672878</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CHR1</td>
<td align="left">CHR1</td>
<td align="left">CHR1</td>
<td align="left">CHR1</td>
<td align="left">CHR1</td>
<td align="left">CHR1</td>
<td align="left">CHR1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.00</td>
<td align="left">0.53</td>
<td align="left">1.05</td>
<td align="left">1.59</td>
<td align="left">2.68</td>
<td align="left">3.20</td>
<td align="left">3.20</td>
</tr>
<tr class="odd">
<td align="left">7046.4863</td>
<td align="left">89.0997</td>
<td align="left">79.57</td>
<td align="left">111.6102</td>
<td align="left">1.25</td>
<td align="left">128.2565</td>
<td align="left">41.44582176</td>
<td align="left">2001-F2-Beta-A10</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
</tr>
<tr class="even">
<td align="left">5883.9174</td>
<td align="left">72.3297</td>
<td align="left">72.12</td>
<td align="left">105.1587</td>
<td align="left">1.45</td>
<td align="left">108.9967</td>
<td align="left">49.62657071</td>
<td align="left">2002-F2-Beta-B10</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
</tr>
<tr class="odd">
<td align="left">8650.5549</td>
<td align="left">88.8501</td>
<td align="left">60.36</td>
<td align="left">149.8485</td>
<td align="left">1.69</td>
<td align="left">124.2265</td>
<td align="left">57.37154613</td>
<td align="left">2003-F2-Beta-C10</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
</tr>
<tr class="even">
<td align="left">8809.2671</td>
<td align="left">86.2728</td>
<td align="left">85.23</td>
<td align="left">135.5865</td>
<td align="left">1.57</td>
<td align="left">197.8657</td>
<td align="left">44.44103452</td>
<td align="left">2004-F2-Beta-D10</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">A</td>
</tr>
</tbody>
</table>
<p>If you look through the full <code>rqtl.csv</code> file, you’ll
notice a few individuals have phenotypic data but no genotypic data.
This is expected—some didn’t meet the quality criteria for genotyping,
were duplicates, or were recorded but not actually sequenced. No
problem: <code>format_input_qtl</code> handled this by assigning
<code>NAs</code>, which rqtl will automatically ignore.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
