<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>QTL analysis using R/qtl and help from geneticMapR • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="QTL analysis using R/qtl and help from geneticMapR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../vignettes/Recode.html">Recode Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/LD_binning.html">Binning SNP Markers</a></li>
    <li><a class="dropdown-item" href="../vignettes/Linkage_Groups.html">Linkage Groups and Marker Order</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno.html">Phenotype Part I</a></li>
    <li><a class="dropdown-item" href="../vignettes/pheno_part_2.html">Phenotype Part II</a></li>
    <li><a class="dropdown-item" href="../vignettes/qtl_analysis.html">Linkage Groups and Marker Order</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>QTL analysis using R/qtl and help from geneticMapR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/vignettes/qtl_analysis.Rmd" class="external-link"><code>vignettes/qtl_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>qtl_analysis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p>This vignette demonstrates a complete workflow for QTL analysis using
the <code>R/qtl</code> package and custom functions from
<code>geneticMapR</code>. It includes genome scans, permutation testing,
composite interval mapping, effect plot generation, and estimation of
percent variance explained (PVE).</p>
</div>
<div class="section level2">
<h2 id="load-packages">Load Packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rqtl.org" class="external-link">qtl</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegaalfaro/geneticMapR" class="external-link">geneticMapR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-and-prepare-data">Load and Prepare Data<a class="anchor" aria-label="anchor" href="#load-and-prepare-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">M2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/read.cross.html" class="external-link">read.cross</a></span><span class="op">(</span><span class="st">"csv"</span>, <span class="st">"."</span>, <span class="st">"../data/rqtl.csv"</span>,</span>
<span>                genotypes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"H"</span>, <span class="st">"B"</span><span class="op">)</span>, alleles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>,</span>
<span>                na.strings <span class="op">=</span> <span class="st">"-"</span>, estimate.map <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;  --Read the following data:</span></span>
<span><span class="co">#&gt;   107  individuals</span></span>
<span><span class="co">#&gt;   950  markers</span></span>
<span><span class="co">#&gt;   8  phenotypes</span></span>
<span><span class="co">#&gt;  --Cross type: f2</span></span>
<span><span class="va">M2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/jittermap.html" class="external-link">jittermap</a></span><span class="op">(</span><span class="va">M2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">M2</span><span class="op">)</span></span>
<span><span class="co">#&gt;     F2 intercross</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     No. individuals:    107 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     No. phenotypes:     8 </span></span>
<span><span class="co">#&gt;     Percent phenotyped: 100 100 100 100 100 100 100 100 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     No. chromosomes:    9 </span></span>
<span><span class="co">#&gt;         Autosomes:      1 2 3 4 5 6 7 8 9 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Total markers:      950 </span></span>
<span><span class="co">#&gt;     No. markers:        100 79 70 106 99 129 163 144 60 </span></span>
<span><span class="co">#&gt;     Percent genotyped:  93.5 </span></span>
<span><span class="co">#&gt;     Genotypes (%):      AA:24.8  AB:54.7  BB:20.5  not BB:0.0  not AA:0.0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">M2</span><span class="op">$</span><span class="va">geno</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "1" "2" "3" "4" "5" "6" "7" "8" "9"</span></span>
<span><span class="va">response_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">M2</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">8</span><span class="op">]</span>  <span class="co"># Remove ID colum</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="calculate-genotype-probabilities-and-visualize-map">Calculate Genotype Probabilities and Visualize Map<a class="anchor" aria-label="anchor" href="#calculate-genotype-probabilities-and-visualize-map"></a>
</h2>
<p>Below, we used a a <code>ggplot2</code> version of the map coverage
visualization named <code>plotMap</code> from geneticMapR. It works on a
<code>scan.one</code> object, the data format from R/qtl.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">M2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/calc.genoprob.html" class="external-link">calc.genoprob</a></span><span class="op">(</span><span class="va">M2</span>, step <span class="op">=</span> <span class="fl">0.25</span>, error.prob <span class="op">=</span> <span class="fl">0.05</span>, map.function <span class="op">=</span> <span class="st">"kosambi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotMap_gg.html">plotMap_gg</a></span><span class="op">(</span><span class="va">M2</span>, horizontal <span class="op">=</span> <span class="cn">TRUE</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="qtl_analysis_files/figure-html/genoprob-map-1.png" width="528"></p>
<div class="section level3">
<h3 id="genetic-map-statistics">Genetic Map Statistics<a class="anchor" aria-label="anchor" href="#genetic-map-statistics"></a>
</h3>
<p>Rqtl estimates statistics of your genetic map which can be calculated
as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's look at the genetic map statistics</span></span>
<span><span class="va">mapStats2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/summary.map.html" class="external-link">summaryMap</a></span><span class="op">(</span><span class="va">M2</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mapStats2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Genetic Map Statistics"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Genetic Map Statistics</caption>
<thead><tr class="header">
<th align="right">n.mar</th>
<th align="right">length</th>
<th align="right">ave.spacing</th>
<th align="right">max.spacing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">100</td>
<td align="right">64.87010</td>
<td align="right">0.6552535</td>
<td align="right">7.660001</td>
</tr>
<tr class="even">
<td align="right">79</td>
<td align="right">58.54008</td>
<td align="right">0.7505138</td>
<td align="right">7.940001</td>
</tr>
<tr class="odd">
<td align="right">70</td>
<td align="right">47.28007</td>
<td align="right">0.6852184</td>
<td align="right">8.160001</td>
</tr>
<tr class="even">
<td align="right">106</td>
<td align="right">67.93010</td>
<td align="right">0.6469534</td>
<td align="right">5.520001</td>
</tr>
<tr class="odd">
<td align="right">99</td>
<td align="right">79.93010</td>
<td align="right">0.8156132</td>
<td align="right">11.090001</td>
</tr>
<tr class="even">
<td align="right">129</td>
<td align="right">75.29013</td>
<td align="right">0.5882041</td>
<td align="right">6.160001</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="permutation-tests">Permutation Tests<a class="anchor" aria-label="anchor" href="#permutation-tests"></a>
</h2>
<p>Before identifying significant QTLs, we determine a LOD (Logarithm of
the Odds) score threshold that accounts for the multiple testing problem
across the genome. A common way to do this is through a permutation
test.</p>
<div class="section level4">
<h4 id="what-is-a-permutation-test">What is a permutation test?<a class="anchor" aria-label="anchor" href="#what-is-a-permutation-test"></a>
</h4>
<p>In QTL mapping, a permutation test is used to empirically estimate
the distribution of maximum LOD scores under the null hypothesis of no
QTL. This is done by randomly shuffling the phenotype values across
individuals and recalculating the genome scan multiple times. By
repeating this process, we generate a distribution of LOD scores
expected by chance, which allows us to establish trait-specific
significance thresholds.</p>
</div>
<div class="section level4">
<h4 id="why-do-we-use-it">Why do we use it?<a class="anchor" aria-label="anchor" href="#why-do-we-use-it"></a>
</h4>
<p>Because many positions across the genome are tested for association
with the trait, the risk of false positives increases. A permutation
test accounts for this genome-wide multiple testing and provides a more
reliable, data-driven threshold for identifying significant QTLs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># number of permutations</span></span>
<span><span class="va">n_perm</span> <span class="op">&lt;-</span> <span class="fl">1000</span> </span>
<span></span>
<span><span class="co"># Initialize an empty list to store the permutation results for each trait</span></span>
<span><span class="va">scanone_results2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop through each response variable (phenotypic trait)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">var</span> <span class="kw">in</span> <span class="va">response_vars</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Run a genome-wide scan using the Haley-Knott regression method with 1000 permutations</span></span>
<span>  <span class="co"># This estimates the LOD threshold for significance for the trait 'var'</span></span>
<span>  <span class="va">perm_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/scanone.html" class="external-link">scanone</a></span><span class="op">(</span><span class="va">M2</span>, pheno.col <span class="op">=</span> <span class="va">var</span>, method <span class="op">=</span> <span class="st">"hk"</span>, n.perm <span class="op">=</span> <span class="va">n_perm</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Summarize the permutation result (returns LOD threshold statistics)</span></span>
<span>  <span class="co"># and store it in the list using the trait name as the key</span></span>
<span>  <span class="va">scanone_results2</span><span class="op">[[</span><span class="va">var</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">perm_result</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span><span class="co">#&gt; Doing permutation in batch mode ...</span></span>
<span></span>
<span><span class="co"># Assign trait names to the elements of the result list</span></span>
<span><span class="co"># This ensures that each element can be identified by the corresponding trait name</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">scanone_results2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">response_vars</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="genomewide-qtl-scan">Genomewide QTL scan<a class="anchor" aria-label="anchor" href="#genomewide-qtl-scan"></a>
</h2>
<p>Below we can start the genome scan. Note that this code is set up to
work with multiple traits. That’s why it uses loops.</p>
<p>Once the LOD significance thresholds are estimated using permutation
tests, the next step is to perform genome-wide QTL scans for each trait
to detect regions associated with phenotypic variation.</p>
<div class="section level4">
<h4 id="what-are-we-doing-here">What are we doing here?<a class="anchor" aria-label="anchor" href="#what-are-we-doing-here"></a>
</h4>
<p>In the code chunk below, we loop through each trait (stored in
response_vars) and apply the scanone() function to perform a genome-wide
scan for QTLs using Haley-Knott regression (method = “hk”), a standard
method for interval mapping in R/qtl.</p>
</div>
<div class="section level4">
<h4 id="why-this-matters">Why this matters?<a class="anchor" aria-label="anchor" href="#why-this-matters"></a>
</h4>
<p>The QTL scan calculates LOD scores across the genome. These scores
quantify the evidence for a QTL at each genomic position. Higher LOD
values suggest a stronger association between the genotype at that locus
and the trait being measured.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Initialize an empty list to store genome scan results for each trait</span></span>
<span><span class="va">scanone_qtl_results2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop through each response variable (i.e., phenotype or trait of interest)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">var</span> <span class="kw">in</span> <span class="va">response_vars</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Perform a genome-wide QTL scan using Haley-Knott regression</span></span>
<span>  <span class="co"># This computes LOD scores across the genome for the given trait</span></span>
<span>  <span class="va">qtl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/scanone.html" class="external-link">scanone</a></span><span class="op">(</span><span class="va">M2</span>, pheno.col <span class="op">=</span> <span class="va">var</span>, method <span class="op">=</span> <span class="st">"hk"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Store the result (a data.frame of LOD scores by chromosome and position) in the list</span></span>
<span>  <span class="va">scanone_qtl_results2</span><span class="op">[[</span><span class="va">var</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">qtl</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>We have made our first genome scan! we can plot the results of the
QTL scan for each trait. This is a good first approximation and we can
look at the results to continue to refine our analysis.</p>
<p>Below we visualize the LOD score profiles to identify regions where
QTLs may be located. Please note that these figures are not publication
ready and <code>geneticMapR</code> has functions to help you create a
better QTL trace plot see further articles on how to do this.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Loop over the list of QTL scan results for each trait</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">scanone_qtl_results2</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Plot the LOD score trace for the i-th trait</span></span>
<span>  <span class="co"># The plot shows LOD scores across the genome</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">scanone_qtl_results2</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, </span>
<span>       main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">scanone_qtl_results2</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">"Pop2"</span><span class="op">)</span><span class="op">)</span>  <span class="co"># Add trait name + "Pop2" as the title</span></span>
<span>  </span>
<span>  <span class="co"># Add a horizontal line representing the 5% significance threshold (LOD cutoff)</span></span>
<span>  <span class="co"># This threshold was previously obtained via permutation test and stored in scanone_results2</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="va">scanone_results2</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, lty <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span>  <span class="co"># Dotted line for visual distinction</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="qtl_analysis_files/figure-html/plot-scanone-1.png" width="528"><img src="qtl_analysis_files/figure-html/plot-scanone-2.png" width="528"><img src="qtl_analysis_files/figure-html/plot-scanone-3.png" width="528"><img src="qtl_analysis_files/figure-html/plot-scanone-4.png" width="528"><img src="qtl_analysis_files/figure-html/plot-scanone-5.png" width="528"><img src="qtl_analysis_files/figure-html/plot-scanone-6.png" width="528"><img src="qtl_analysis_files/figure-html/plot-scanone-7.png" width="528"></p>
</div>
</div>
<div class="section level2">
<h2 id="composite-interval-mapping-cim">Composite Interval Mapping (CIM)<a class="anchor" aria-label="anchor" href="#composite-interval-mapping-cim"></a>
</h2>
<p>To improve QTL detection and reduce the effects of background genetic
noise, we now perform Composite Interval Mapping (CIM).</p>
<div class="section level4">
<h4 id="what-is-cim">What is CIM?<a class="anchor" aria-label="anchor" href="#what-is-cim"></a>
</h4>
<p>CIM is an extension of interval mapping that improves precision and
power by including background markers (covariates) in the model. These
covariates help account for genetic variation from loci outside the
interval being tested, reducing false positives and improving detection
of closely linked QTLs.</p>
<p>CIM helps to:</p>
<ul>
<li><p>Control for the genetic background, reducing confounding
effects</p></li>
<li><p>Improve resolution to detect QTLs</p></li>
<li><p>Differentiate between closely linked QTLs</p></li>
</ul>
<p>From the previous QTL trace plots, it looks like there are 4 or 5
peaks for length to width ratio, so we can create a look up table with
this information. No other significant signals were identified for the
remaining traits.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a lookup table that assigns the number of covariates (marcovars) </span></span>
<span><span class="co"># to use for each trait during composite interval mapping</span></span>
<span><span class="va">lookup_table2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  var <span class="op">=</span> <span class="va">response_vars</span>,                <span class="co">#Vector of trait names</span></span>
<span>  covar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span>      <span class="co">#Corresponding number of covariates for each trait</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Helper function to retrieve the number of covariates for a given trait</span></span>
<span><span class="va">get_n_covar</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">var</span>, <span class="va">lookup_table2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">var</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">lookup_table2</span><span class="op">$</span><span class="va">var</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># If the trait is listed in the lookup table, return its associated covariate number</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">lookup_table2</span><span class="op">$</span><span class="va">covar</span><span class="op">[</span><span class="va">lookup_table2</span><span class="op">$</span><span class="va">var</span> <span class="op">==</span> <span class="va">var</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="co"># If the trait is not listed, return an error</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"No covariate info for"</span>, <span class="va">var</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Initialize an empty list to store the CIM results for each trait</span></span>
<span><span class="va">cim_qtl_results2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop over each response variable (trait)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">var</span> <span class="kw">in</span> <span class="va">response_vars</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Retrieve the number of covariates to use for this trait</span></span>
<span>  <span class="va">n_marcovar</span> <span class="op">&lt;-</span> <span class="fu">get_n_covar</span><span class="op">(</span><span class="va">var</span>, <span class="va">lookup_table2</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Perform composite interval mapping using the Haley-Knott regression method</span></span>
<span>  <span class="co"># pheno.col = var: Use the current trait as the phenotype</span></span>
<span>  <span class="co"># n.marcovar: Number of background markers to use as covariates</span></span>
<span>  <span class="co"># window = 10: Exclude markers within 10 cM of the test interval to avoid collinearity</span></span>
<span>  <span class="co"># map.function = "kosambi": Use Kosambi mapping function for converting recombination to distance</span></span>
<span>  <span class="va">qtl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/cim.html" class="external-link">cim</a></span><span class="op">(</span><span class="va">M2</span>, pheno.col <span class="op">=</span> <span class="va">var</span>, method <span class="op">=</span> <span class="st">"hk"</span>, n.marcovar <span class="op">=</span> <span class="va">n_marcovar</span>,</span>
<span>             window <span class="op">=</span> <span class="fl">10</span>, map.function <span class="op">=</span> <span class="st">"kosambi"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Store the CIM result in the list, by trait name</span></span>
<span>  <span class="va">cim_qtl_results2</span><span class="op">[[</span><span class="va">var</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">qtl</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="permutation-test-for-cim">Permutation test for CIM<a class="anchor" aria-label="anchor" href="#permutation-test-for-cim"></a>
</h3>
<p>As with simple interval mapping, composite interval mapping also
requires empirical significance thresholds to determine whether a QTL is
statistically significant. This is achieved through permutation testing,
which simulates the null distribution of LOD scores.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Initialize an empty list to store  results </span></span>
<span><span class="va">cim_perm_pop2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop through each  trait</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">var</span> <span class="kw">in</span> <span class="va">response_vars</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># get the no. of covariates (background markers) to include for this trait</span></span>
<span>  <span class="va">n_marcovar</span> <span class="op">&lt;-</span> <span class="fu">get_n_covar</span><span class="op">(</span><span class="va">var</span>, <span class="va">lookup_table2</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Perform a permutation test using CIM </span></span>
<span>  <span class="co">#n.perm = 100: Number of permutations (typically 100–1000 adjust based on time/resources)</span></span>
<span>  <span class="co"># - method = "hk": Use Haley-Knott regression</span></span>
<span>  <span class="co"># - map.function = "kosambi": Use Kosambi map function for genetic distance</span></span>
<span>  <span class="co">#- n.marcovar = n_marcovar: Number of background markers as covariates</span></span>
<span>  <span class="co"># - window = 10: Exclude markers within 10 cM of the test interval</span></span>
<span>  <span class="co"># - error.prob = 0.05: Default genotyping error probability</span></span>
<span>  <span class="va">perm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qtl/man/cim.html" class="external-link">cim</a></span><span class="op">(</span><span class="va">M2</span>, pheno.col <span class="op">=</span> <span class="va">var</span>, method <span class="op">=</span> <span class="st">"hk"</span>, n.perm <span class="op">=</span> <span class="fl">100</span>,</span>
<span>              map.function <span class="op">=</span> <span class="st">"kosambi"</span>, n.marcovar <span class="op">=</span> <span class="va">n_marcovar</span>,</span>
<span>              window <span class="op">=</span> <span class="fl">10</span>, error.prob <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Store the summarized permutation results (e.g., LOD thresholds) in the list</span></span>
<span>  <span class="va">cim_perm_pop2</span><span class="op">[[</span><span class="va">var</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">perm</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Assign trait names as keys for easy identification</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cim_perm_pop2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">response_vars</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plotting-cim">Plotting CIM<a class="anchor" aria-label="anchor" href="#plotting-cim"></a>
</h3>
<p>After performing Composite Interval Mapping (CIM) and estimating
trait-specific significance thresholds through permutation testing, we
visualize the LOD score profiles to identify significant QTLs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">"Pop2 CIM"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="va">cim_perm_pop2</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, lty <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="qtl_analysis_files/figure-html/plot-cim-1.png" width="528"><img src="qtl_analysis_files/figure-html/plot-cim-2.png" width="528"><img src="qtl_analysis_files/figure-html/plot-cim-3.png" width="528"><img src="qtl_analysis_files/figure-html/plot-cim-4.png" width="528"><img src="qtl_analysis_files/figure-html/plot-cim-5.png" width="528"><img src="qtl_analysis_files/figure-html/plot-cim-6.png" width="528"><img src="qtl_analysis_files/figure-html/plot-cim-7.png" width="528"></p>
<p>Please notice that the plot for length-to-width ratio using CIM is
different from the plot from our previous regular genome scan. The peaks
seem clearer and narrower when CIM was used.</p>
</div>
</div>
<div class="section level2">
<h2 id="summary-of-significant-qtls">Summary of Significant QTLs<a class="anchor" aria-label="anchor" href="#summary-of-significant-qtls"></a>
</h2>
<p>We can save the results of the QTLs scan in a list and query them
later.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results_pop2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, threshold <span class="op">=</span> <span class="va">cim_perm_pop2</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results_pop2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="lod-support-interval">LOD Support Interval<a class="anchor" aria-label="anchor" href="#lod-support-interval"></a>
</h2>
<p>To estimate the genomic region likely to contain a QTL, we can
calculate the LOD support interval. This interval is defined as the
region around a QTL peak where the LOD score drops by a specified amount
(commonly 1.5 units) from its maximum. This approach helps to define
confidence intervals for QTL positions and assists in narrowing down
candidate gene regions.</p>
<p>In the example below, we use the <code><a href="../reference/get_LOD_int.html">get_LOD_int()</a></code> function
to extract 1.5-LOD drop support intervals for the trait
length_width_ratio using the cim_qtl_results2 object, the genetic map
M2, and the corresponding population results:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lw_result_pop2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_LOD_int.html">get_LOD_int</a></span><span class="op">(</span><span class="va">M2</span>, <span class="va">cim_qtl_results2</span>, <span class="va">results_pop2</span>,</span>
<span>                              trait <span class="op">=</span> <span class="st">"length_width_ratio"</span>, drop <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p>The output, lw_result_pop2, is a data frame listing the start and end
positions of the LOD support intervals for each significant QTL. These
intervals provide useful bounds for downstream analyses, such as
identifying candidate genes or comparing results across populations.</p>
</div>
<div class="section level2">
<h2 id="effect-plots">Effect Plots<a class="anchor" aria-label="anchor" href="#effect-plots"></a>
</h2>
<p>We can visualize the allelic effects of a specific QTL using the
<code><a href="../reference/effect_plot.html">effect_plot()</a></code> function. The effect plot that I propose
below is powerful as it shows the distribution, the median and the
number of individuals under each category. Below, we load the effect
estimates and generate a plot for QTL_1 associated with the
length_width_ratio trait:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">effects2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"../data/rqtl.csv"</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Let's Define the marker of interest for QTL_1</span></span>
<span><span class="va">marker1</span> <span class="op">&lt;-</span> <span class="va">lw_result_pop2</span><span class="op">$</span><span class="va">QTL_1</span><span class="op">$</span><span class="va">usable_marker</span></span>
<span></span>
<span><span class="co"># Let's create an effect plot with a default with flipped coords</span></span>
<span><span class="fu"><a href="../reference/effect_plot.html">effect_plot</a></span><span class="op">(</span><span class="va">effects2</span>, <span class="va">marker1</span>, trait_name <span class="op">=</span> <span class="st">"length_width_ratio"</span>,</span>
<span>                trait_label <span class="op">=</span> <span class="st">"Length-to-width ratio"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="qtl_analysis_files/figure-html/effect-plots-1.png" width="528"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 6</span></span></span>
<span><span class="co">#&gt;   CHR4_12690811 median   max   min count    sd</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> A               1.88  2.56  1.21    18 0.452</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> H               1.52  2.22  0.79    56 0.344</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> B               1.18  2.1   0.79    26 0.374</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;   pop_sd pop_mean pop_median pop_max</span></span>
<span><span class="co">#&gt; 1   0.42     1.52       1.48    2.56</span></span></code></pre>
<p>This plot shows the mean trait values for each genotype at the
marker, helping interpret the direction and magnitude of the QTL
effect.</p>
<p>Repeat for other markers as needed as the workflow makes it really
easy to obtain effect plots for your QTLs!</p>
<p>Notice that below the plot, the function effec_plot will output
phenotypic summary statistics for each of the genotype categories and
for the whole population of individuals!</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Let's Define the marker of interest for QTL_1</span></span>
<span><span class="va">marker2</span> <span class="op">&lt;-</span> <span class="va">lw_result_pop2</span><span class="op">$</span><span class="va">QTL_2</span><span class="op">$</span><span class="va">usable_marker</span></span>
<span></span>
<span><span class="co"># Let's create an effect plot with a default with flipped coords</span></span>
<span><span class="fu"><a href="../reference/effect_plot.html">effect_plot</a></span><span class="op">(</span><span class="va">effects2</span>, <span class="va">marker2</span>, trait_name <span class="op">=</span> <span class="st">"length_width_ratio"</span>,</span>
<span>                trait_label <span class="op">=</span> <span class="st">"Length-to-width ratio"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="qtl_analysis_files/figure-html/effect-plots2-1.png" width="528"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 6</span></span></span>
<span><span class="co">#&gt;   CHR6_8692961 median   max   min count    sd</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> A              1.85  2.56  1.17    25 0.391</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> H              1.46  2.5   0.79    44 0.390</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> B              1.3   2.27  0.79    31 0.366</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt;   pop_sd pop_mean pop_median pop_max</span></span>
<span><span class="co">#&gt; 1   0.42     1.52       1.48    2.56</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="percent-variance-explained-pve">Percent Variance Explained (PVE)<a class="anchor" aria-label="anchor" href="#percent-variance-explained-pve"></a>
</h2>
<p>After identifying significant QTLs, we can quantify how much of the
phenotypic variation each QTL accounts for. This is typically done by
calculating the Percent Variance Explained using a simple linear
model.</p>
<div class="section level4">
<h4 id="but-what-is-pve">But what is PVE?<a class="anchor" aria-label="anchor" href="#but-what-is-pve"></a>
</h4>
<p>PVE represents the proportion of variation in a trait that can be
attributed to a specific marker or QTL. It helps assess the effect size
of the QTL and how important it is in controlling the trait.</p>
</div>
<div class="section level4">
<h4 id="interpretation">Interpretation<a class="anchor" aria-label="anchor" href="#interpretation"></a>
</h4>
<p>The R-squared value from the linear model indicates the proportion of
variance in length_width_ratio explained by the marker marker1.</p>
<p>To report this as PVE, simply multiply the “Multiple R-squared” value
by 100.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># CHR4 example</span></span>
<span><span class="co"># Select the marker name of interest, for example,a significant QTL on chromosome 4</span></span>
<span><span class="va">colname</span> <span class="op">&lt;-</span> <span class="va">marker1</span>  <span class="co"># This should be a character string like "CHR4_123456"</span></span>
<span></span>
<span><span class="co"># Extract the genotype data for this marker from the effects2 dataset</span></span>
<span><span class="va">effects2</span><span class="op">$</span><span class="va">sig_marker1_pop2</span> <span class="op">&lt;-</span> <span class="va">effects2</span><span class="op">[[</span><span class="va">colname</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Fit a linear model with the phenotype as the response and marker genotype as the predictor</span></span>
<span><span class="va">fit_CHR4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">length_width_ratio</span> <span class="op">~</span> <span class="fl">1</span> <span class="op">+</span> <span class="va">sig_marker1_pop2</span>, data <span class="op">=</span> <span class="va">effects2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the model summary, which includes the R-squared value used to calculate PVE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fit_CHR4</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; lm(formula = length_width_ratio ~ 1 + sig_marker1_pop2, data = effects2)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residuals:</span></span>
<span><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span><span class="co">#&gt; -0.72339 -0.30308 -0.03225  0.23661  0.81692 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt;                   Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">#&gt; (Intercept)        1.88111    0.08792  21.396  &lt; 2e-16 ***</span></span>
<span><span class="co">#&gt; sig_marker1_pop2B -0.59803    0.11437  -5.229 9.81e-07 ***</span></span>
<span><span class="co">#&gt; sig_marker1_pop2H -0.36772    0.10107  -3.638 0.000442 ***</span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residual standard error: 0.373 on 97 degrees of freedom</span></span>
<span><span class="co">#&gt;   (7 observations deleted due to missingness)</span></span>
<span><span class="co">#&gt; Multiple R-squared:  0.2201, Adjusted R-squared:  0.204 </span></span>
<span><span class="co">#&gt; F-statistic: 13.69 on 2 and 97 DF,  p-value: 5.802e-06</span></span></code></pre></div>
<p>The QTL on chromosome 4 shown in the first effect plot and in the PVE
example explains 21.66% of the phenotypic variance if we look at the
multiple R-squared.</p>
</div>
</div>
<div class="section level2">
<h2 id="save-results">Save Results<a class="anchor" aria-label="anchor" href="#save-results"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">cim_qtl_results2</span>, </span>
<span>     <span class="va">cim_perm_pop2</span>, </span>
<span>     <span class="va">M2</span>, </span>
<span>     <span class="va">effects2</span>,</span>
<span>     <span class="va">mapStats2</span>,</span>
<span>     <span class="va">scanone_qtl_results2</span>, </span>
<span>     <span class="va">scanone_results2</span>,</span>
<span>     file <span class="op">=</span> <span class="st">"../data/plot_trace_pop2.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This vignette provides a streamlined and reproducible QTL mapping
workflow using R/qtl and custom plotting utilities from genticMapR. The
steps shown allow detection, validation, of QTLs in an efficient manner.
Go to the next article to learn more about better visualizing the QTL
results using functions from <code>geneticMapR.</code></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
