<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Recode Markers • geneticMapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Recode Markers">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geneticMapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/geneticMapR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Recode.html">Recode Markers</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/vegaalfaro/geneticMapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Recode Markers</h1>
                        <h4 data-toc-skip class="author">Andrey Vega
Alfaro</h4>
            
            <h4 data-toc-skip class="date">2025-04-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vegaalfaro/geneticMapR/blob/master/vignettes/Recode.Rmd" class="external-link"><code>vignettes/Recode.Rmd</code></a></small>
      <div class="d-none name"><code>Recode.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This article demonstrates how to process, clean, and recode (phase)
SNP markers from a VCF file to generate genotype matrices compatible
with genetic mapping workflows. We will use helper functions from the
<code>geneticMapR</code> package as well as dependencies such as
<code>MapRtools</code>, <code>VariantAnnotation</code>, and
<code>Rsamtools</code>. The example focuses on an F2 population, where
genotype recoding is based on founder descent.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>We will need a series of packages, below we show how to install them
if needed.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Install required packages (if not already installed)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"geneticMapR"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"vegaalfaro/geneticMapR"</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"MapRtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"jendelman/MapRtools"</span><span class="op">)</span></span>
<span><span class="co">#&gt; gmp          (NA -&gt; 0.7-5   ) [CRAN]</span></span>
<span><span class="co">#&gt; bit          (NA -&gt; 4.6.0   ) [CRAN]</span></span>
<span><span class="co">#&gt; permute      (NA -&gt; 0.9-7   ) [CRAN]</span></span>
<span><span class="co">#&gt; iterators    (NA -&gt; 1.0.14  ) [CRAN]</span></span>
<span><span class="co">#&gt; foreach      (NA -&gt; 1.5.2   ) [CRAN]</span></span>
<span><span class="co">#&gt; RcppArmad... (NA -&gt; 14.4.1-1) [CRAN]</span></span>
<span><span class="co">#&gt; osqp         (NA -&gt; 0.6.3.3 ) [CRAN]</span></span>
<span><span class="co">#&gt; scs          (NA -&gt; 3.2.7   ) [CRAN]</span></span>
<span><span class="co">#&gt; ECOSolveR    (NA -&gt; 0.5.5   ) [CRAN]</span></span>
<span><span class="co">#&gt; Rmpfr        (NA -&gt; 1.0-0   ) [CRAN]</span></span>
<span><span class="co">#&gt; bit64        (NA -&gt; 4.6.0-1 ) [CRAN]</span></span>
<span><span class="co">#&gt; vegan        (NA -&gt; 2.6-10  ) [CRAN]</span></span>
<span><span class="co">#&gt; TSP          (NA -&gt; 1.2-4   ) [CRAN]</span></span>
<span><span class="co">#&gt; registry     (NA -&gt; 0.5-1   ) [CRAN]</span></span>
<span><span class="co">#&gt; qap          (NA -&gt; 0.1-2   ) [CRAN]</span></span>
<span><span class="co">#&gt; gclus        (NA -&gt; 1.3.3   ) [CRAN]</span></span>
<span><span class="co">#&gt; ca           (NA -&gt; 0.71.1  ) [CRAN]</span></span>
<span><span class="co">#&gt; splines2     (NA -&gt; 0.5.4   ) [CRAN]</span></span>
<span><span class="co">#&gt; CVXR         (NA -&gt; 1.0-15  ) [CRAN]</span></span>
<span><span class="co">#&gt; seriation    (NA -&gt; 1.5.7   ) [CRAN]</span></span>
<span><span class="co">#&gt; scam         (NA -&gt; 1.2-18  ) [CRAN]</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="color: #00BBBB;">R CMD build</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; * checking for file ‘/tmp/Rtmpzy1uDt/remotes1ec325f327c2/jendelman-MapRtools-cabbd53/DESCRIPTION’ ... OK</span></span>
<span><span class="co">#&gt; * preparing ‘MapRtools’:</span></span>
<span><span class="co">#&gt; * checking DESCRIPTION meta-information ... OK</span></span>
<span><span class="co">#&gt; * checking for LF line-endings in source and make files and shell scripts</span></span>
<span><span class="co">#&gt; * checking for empty or unneeded directories</span></span>
<span><span class="co">#&gt; Omitted ‘LazyData’ from DESCRIPTION</span></span>
<span><span class="co">#&gt; * building ‘MapRtools_0.36.tar.gz’</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"VariantAnnotation"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"VariantAnnotation"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"Rsamtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"Rsamtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span></span>
<span><span class="co"># Load libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegaalfaro/geneticMapR" class="external-link">geneticMapR</a></span><span class="op">)</span> </span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MapRtools</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Rsamtools" class="external-link">Rsamtools</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-and-explore-vcf-file">Load and Explore VCF File<a class="anchor" aria-label="anchor" href="#load-and-explore-vcf-file"></a>
</h2>
<p>The VCF file is somewhat large (19.8 MB) to be included as a system
file. The repository <a href="https://github.com/vegaalfaro/geneticMapRFiles" class="external-link">geneticMapRFiles</a>
is a data-only repository that will help us save data and run through
examples without much issues. Let’s get a local copy of the VCF file
using <code><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">utils::download.file()</a></code></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's define our URL from the data repository that accompanies this R package</span></span>
<span><span class="va">vcf_url</span> <span class="op">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/vegaalfaro/geneticMapRFiles/main/vcf/SNP_updated_IDs_sorted2.vcf.gz"</span></span>
<span></span>
<span><span class="co"># Download file</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"local_copy.vcf.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">vcf_url</span>, destfile <span class="op">=</span> <span class="st">"local_copy.vcf.gz"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>The code below loads the vcf file using the function readVcf.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's read the vcf into our environment</span></span>
<span><span class="va">vcf2</span> <span class="op">&lt;-</span> <span class="fu">VariantAnnotation</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html" class="external-link">readVcf</a></span><span class="op">(</span><span class="st">"local_copy.vcf.gz"</span>, genome <span class="op">=</span> <span class="st">"unknown"</span><span class="op">)</span></span></code></pre></div>
<p>Let’s explore the vcf file. The VCF contains several fields including
GT (genotype), DP (depth), GQ (genotype quality) and PL (sample-level
annotations)</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Overview</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">geno</a></span><span class="op">(</span><span class="va">vcf2</span><span class="op">)</span>, <span class="va">class</span><span class="op">)</span></span>
<span><span class="co">#&gt;      GT       AD       DP       GQ       PL      </span></span>
<span><span class="co">#&gt; [1,] "matrix" "matrix" "matrix" "matrix" "matrix"</span></span>
<span><span class="co">#&gt; [2,] "array"  "array"  "array"  "array"  "array"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="extract-genotype-and-coverage-information">Extract Genotype and Coverage Information<a class="anchor" aria-label="anchor" href="#extract-genotype-and-coverage-information"></a>
</h2>
<p>Let’s extract the genotype and depth which are saved in the GT and DP
fields using VariantAnnotation’s geno function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">geno</a></span><span class="op">(</span><span class="va">vcf2</span><span class="op">)</span><span class="op">$</span><span class="va">GT</span></span>
<span><span class="va">DP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">geno</a></span><span class="op">(</span><span class="va">vcf2</span><span class="op">)</span><span class="op">$</span><span class="va">DP</span></span></code></pre></div>
<p>Once extracted, we can estimate depth statistics.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">DP</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;    0.00   19.00   27.00   30.84   39.00 1656.00</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="convert-genotype-call-to-dosage">Convert Genotype Call to Dosage<a class="anchor" aria-label="anchor" href="#convert-genotype-call-to-dosage"></a>
</h2>
<p>If we inspect the <code>GT</code> object,</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GT</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;             7001-F1-Beta-H9 2001-F2-Beta-A10 2002-F2-Beta-B10</span></span>
<span><span class="co">#&gt; CHR7_192222 "1/0"           "0/1"            "1/1"           </span></span>
<span><span class="co">#&gt; CHR7_192239 "0/0"           "0/0"            "0/0"           </span></span>
<span><span class="co">#&gt; CHR7_192241 "0/0"           "0/0"            "0/0"</span></span></code></pre></div>
<p>genotypes are coded as <code>"0/0"</code>, <code>"0/1"</code>, or
<code>"1/1"</code>, representing the maximum likelihood call:
<code>0</code> for the reference (REF) allele and <code>1</code> for the
alternate (ALT) allele. For example, <code>"0/0"</code> means a dosage
of 0 ALT alleles, <code>"0/1"</code> or <code>"1/0"</code> is a dosage
of 1 ALT allele and <code>"1/1"</code> corresponds to a dosage of 2.</p>
<p>The <code>convert_to_dosage</code> family of functions in
<strong>geneticMapR</strong> handles this conversion easily. For
polyploids (e.g., <code>"0/0/0/0"</code>), check out
<code>convert_to_dosage_advanced</code> and
<code>convert_to_dosage_flex</code>.</p>
<p>Let’s convert the genotype calls into dosages:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno_1629</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_to_dosage_flex.html">convert_to_dosage_flex</a></span><span class="op">(</span><span class="va">GT</span><span class="op">)</span></span></code></pre></div>
<p>Let’s take a look at the conversion.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno_1629</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;             7001-F1-Beta-H9 2001-F2-Beta-A10 2002-F2-Beta-B10</span></span>
<span><span class="co">#&gt; CHR7_192222               1                1                2</span></span>
<span><span class="co">#&gt; CHR7_192239               0                0                0</span></span>
<span><span class="co">#&gt; CHR7_192241               0                0                0</span></span></code></pre></div>
<p>We have converted to dosage from allele format call.</p>
</div>
<div class="section level2">
<h2 id="filter-markers-and-individuals-by-missing-data">Filter Markers and Individuals by Missing Data<a class="anchor" aria-label="anchor" href="#filter-markers-and-individuals-by-missing-data"></a>
</h2>
<p>The function <code><a href="../reference/filter_missing_geno.html">filter_missing_geno()</a></code> takes a genotype
matrix with rows representing genetic markers and columns representing
individuals. The <code>geno_matrix</code> argument specifies the
genotype matrix.</p>
<p>The <code>threshold</code> argument allows to specify the max
proportion of missing data before an individual or marker is removed.
Ranges from 0 to 1 and the default is 0.10.</p>
<p>Let’s filter by <code>markers</code> first and accept 10% missing
data.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_missing_geno.html">filter_missing_geno</a></span><span class="op">(</span>geno_matrix <span class="op">=</span> <span class="va">geno_1629</span>, </span>
<span>                              threshold <span class="op">=</span> <span class="fl">0.10</span>,</span>
<span>                              filter_by <span class="op">=</span> <span class="st">"markers"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 0 markers removed (Threshold: 0.1)</span></span>
<span></span>
<span><span class="co"># Extract our result</span></span>
<span><span class="va">filtered_geno</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">filtered_geno</span></span></code></pre></div>
<p>We can see that no marker was removed. Our markers have all less than
10% missing data. We can confirm it using a plot.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">missing_values</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">pct_missing</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">missing_values</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"missing"</span></span>
<span></span>
<span><span class="co"># Visualize missingness</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">missing_values</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">missing</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/plot%20missing%20data%20markers-1.png" alt="plot missing" width="585.6" style="display: block; margin: auto;"></p>
<p>We can also filter by missing data per <code>individuals</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno_missing_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_missing_geno.html">filter_missing_geno</a></span><span class="op">(</span><span class="va">filtered_geno</span>,</span>
<span>                                             threshold <span class="op">=</span> <span class="fl">0.10</span>,</span>
<span>                                             filter_by <span class="op">=</span> <span class="st">"individuals"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 7 individuals removed (Threshold: 0.1)</span></span>
<span><span class="va">filtered_geno</span> <span class="op">&lt;-</span> <span class="va">geno_missing_filtered</span><span class="op">$</span><span class="va">filtered_geno</span></span></code></pre></div>
<p>The function created a message letting us know that 7 individuals
were removed as they have data that exceeds the 0.1 threshold.</p>
<p>Let’s see an example on how to access the results.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access outputs</span></span>
<span><span class="va">filtered_geno</span> <span class="op">&lt;-</span> <span class="va">geno_missing_filtered</span><span class="op">$</span><span class="va">filtered_geno</span></span>
<span><span class="va">missing_values</span> <span class="op">&lt;-</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">geno_missing_filtered</span><span class="op">$</span><span class="va">pct_missing</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">missing_values</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"missing"</span></span>
<span><span class="va">removed_individuals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ind <span class="op">=</span> <span class="va">geno_missing_filtered</span><span class="op">$</span><span class="va">removed_individuals</span><span class="op">)</span></span></code></pre></div>
<p>Let’s see which individuals were removed</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">removed_individuals</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 ind</span></span>
<span><span class="co">#&gt; 1  2055-F2-Gamma-G4</span></span>
<span><span class="co">#&gt; 2  2088-F2-Gamma-H8</span></span>
<span><span class="co">#&gt; 3 2097-F2-Gamma-A10</span></span>
<span><span class="co">#&gt; 4 2099-F2-Gamma-C10</span></span>
<span><span class="co">#&gt; 5 2105-F2-Gamma-A11</span></span>
<span><span class="co">#&gt; 6 2106-F2-Gamma-B11</span></span>
<span><span class="co">#&gt; 7 2107-F2-Gamma-C11</span></span></code></pre></div>
<p>We can visualize how much missing data we still have in the kept
individuals. It is no greater than 7% which is in agreement with our
filtering parameters.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize missing data</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">missing_values</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">missing</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/plot%20missing%20data%20individuals-1.png" alt="missing data plot" width="585.6" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="visualize-genotype-data">Visualize Genotype Data<a class="anchor" aria-label="anchor" href="#visualize-genotype-data"></a>
</h2>
<p>We can also plot a histogram of genotypic values using
<code><a href="../reference/plot_genotype_histogram.html">plot_genotype_histogram()</a></code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_genotype_histogram.html">plot_genotype_histogram</a></span><span class="op">(</span><span class="va">filtered_geno</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/viz%20genotype%20data-1.png" width="585.6" style="display: block; margin: auto;"></p>
<p>Let’s also take a look at the coverage of the markers across the
genome and chromosomes. This function is based on MapRtool’s
<code>plot_coverage.</code> I just added a few customization. Check it
out <a href="https://github.com/jendelman/MapRtools/tree/master/R" class="external-link">here</a>.</p>
<p>Because our filtered genotype matrix has the marker names in the row
names in the format “CHR7_192222” we can use the function
<code><a href="../reference/extract_map.html">extract_map()</a></code> from <code>geneticMapR</code> to easily
create a physical map with columns “marker” “chrom” and “position”.</p>
<p>The function <code><a href="../reference/extract_map.html">extract_map()</a></code> is very flexible but the row
names should contain components (chromosome, position) separated by a
symbol (e.g., _, -, .). Use arguments <code>chrom_index</code> and
<code>pos_index</code> to specify their positions, and split_symbol to
define the delimiter.</p>
<p>In our previous example <strong>“CHR7_192222”</strong>,
<code>chrom_index = 1</code> as the chromosome is the first field before
the delimiter “_“. If our markers were coded as
<strong>W257B_“CHR7_192222”</strong> then <code>chrom_index = 2</code>
and <code>pos_index = 3</code> as chromosome and position are in the
second and third field respectively.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract physical map</span></span>
<span><span class="va">map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_map.html">extract_map</a></span><span class="op">(</span>genotype_matrix <span class="op">=</span> <span class="va">filtered_geno</span>,</span>
<span>                   chrom_index <span class="op">=</span> <span class="fl">1</span>, <span class="co"># Index for Chromosome </span></span>
<span>                   pos_index <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                    markers <span class="op">=</span> <span class="cn">FALSE</span>,  <span class="co"># If TRUE, includes original marker names</span></span>
<span>                    split_symbol <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> </span></code></pre></div>
<p>Let’s see the first few lines of the map</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># See map heading</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">map</span><span class="op">)</span></span>
<span><span class="co">#&gt;   chrom position</span></span>
<span><span class="co">#&gt; 1  CHR7   192222</span></span>
<span><span class="co">#&gt; 2  CHR7   192239</span></span>
<span><span class="co">#&gt; 3  CHR7   192241</span></span>
<span><span class="co">#&gt; 4  CHR7   192273</span></span>
<span><span class="co">#&gt; 5  CHR7   225702</span></span>
<span><span class="co">#&gt; 6  CHR7   225710</span></span></code></pre></div>
<p>Now we are ready for plotting:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">geneticMapR</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_coverage.html">plot_coverage</a></span><span class="op">(</span>map<span class="op">=</span><span class="va">map</span>, customize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/plot%20coverage%201-1.png" alt="coverage plot" width="585.6" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="genotype-frequency">Genotype Frequency<a class="anchor" aria-label="anchor" href="#genotype-frequency"></a>
</h2>
<p>We can evaluate homozygosity levels in genotype data using the
<code><a href="../reference/freq.html">freq()</a></code> function. The function computes the relative
frequency of each genotype (typically coded as 0, 1, or 2) across
markers or individuals.</p>
<div class="section level4">
<h4 id="genotype-frequency-by-marker">Genotype Frequency by Marker<a class="anchor" aria-label="anchor" href="#genotype-frequency-by-marker"></a>
</h4>
<p>To evaluate how homozygous the parents (or any sample) are across
markers, we first calculate genotype frequencies by marker:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno.freq.mar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freq.html">freq</a></span><span class="op">(</span><span class="va">filtered_geno</span>, input_format <span class="op">=</span> <span class="st">"numeric"</span>, by <span class="op">=</span> <span class="st">"markers"</span><span class="op">)</span></span></code></pre></div>
<p>This returns the relative frequency of each genotype (0, 1, 2) at
each marker locus across all individuals, allowing identification of
loci with high homozygosity or heterozygosity.</p>
<p>Let’s take a look</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">geno.freq.mar</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 0         1         2</span></span>
<span><span class="co">#&gt; CHR7_192222 0.125 0.5288462 0.3461538</span></span>
<span><span class="co">#&gt; CHR7_192239 1.000 0.0000000 0.0000000</span></span>
<span><span class="co">#&gt; CHR7_192241 1.000 0.0000000 0.0000000</span></span>
<span><span class="co">#&gt; CHR7_192273 1.000 0.0000000 0.0000000</span></span>
<span><span class="co">#&gt; CHR7_225702 1.000 0.0000000 0.0000000</span></span>
<span><span class="co">#&gt; CHR7_225710 1.000 0.0000000 0.0000000</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="genotype-frequency-by-marker-1">Genotype Frequency by Marker<a class="anchor" aria-label="anchor" href="#genotype-frequency-by-marker-1"></a>
</h4>
<p>To evaluate the homozygosity within individuals across all
markers:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno.freq.ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freq.html">freq</a></span><span class="op">(</span><span class="va">filtered_geno</span>, input_format <span class="op">=</span> <span class="st">"numeric"</span>, by <span class="op">=</span> <span class="st">"individuals"</span><span class="op">)</span></span></code></pre></div>
<p>This produces the relative frequency of each genotype for every
individual, helping to pinpoint individuals with unusually high or low
homozygosity levels.</p>
<p>Let’s take a look</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">geno.freq.ind</span><span class="op">)</span></span>
<span><span class="co">#&gt;                          0          1          2</span></span>
<span><span class="co">#&gt; 7001-F1-Beta-H9  0.8822163 0.09591408 0.02186960</span></span>
<span><span class="co">#&gt; 2001-F2-Beta-A10 0.8940523 0.07536372 0.03058399</span></span>
<span><span class="co">#&gt; 2002-F2-Beta-B10 0.9041417 0.05869674 0.03716160</span></span>
<span><span class="co">#&gt; 2003-F2-Beta-C10 0.8998198 0.06497826 0.03520198</span></span>
<span><span class="co">#&gt; 2004-F2-Beta-D10 0.9027175 0.05645933 0.04082315</span></span>
<span><span class="co">#&gt; 2005-F2-Beta-E10 0.8934391 0.07540088 0.03116000</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="marker-type-analysis">Marker Type Analysis<a class="anchor" aria-label="anchor" href="#marker-type-analysis"></a>
</h2>
<p>Markers can be classified based on the genotypes of two parents:</p>
<ul>
<li>
<strong>Homozygous</strong>: P1 = 0 &amp; P2 = 2 (or vice
versa)</li>
<li>
<strong>Non-polymorphic</strong>: Both parents have the same
genotype (e.g., P1 = P2 = 0, 1, or 2)</li>
<li>
<strong>Heterozygous</strong>: One or both parents are heterozygous
(e.g., P1 = 1 &amp; P2 = 0)</li>
</ul>
<p>Non-polymorphic markers should always be removed. Heterozygous
markers can optionally be excluded depending on your analysis goals.</p>
<p>To filter for homozygous markers only use:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Declare the name of the parents as they appear in the column of your geno</span></span>
<span><span class="va">P1</span> <span class="op">&lt;-</span> <span class="st">"P2550-Cylindra-P1-Theta-A9"</span></span>
<span><span class="va">P2</span> <span class="op">&lt;-</span> <span class="st">"P2493-Mono-P2-Theta-B9"</span></span>
<span></span>
<span><span class="co"># Run our function to fiter out heterozygous markers</span></span>
<span><span class="va">geno_homozygous</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_geno_by_parents.html">filter_geno_by_parents</a></span><span class="op">(</span><span class="va">filtered_geno</span>, <span class="va">P1</span>, <span class="va">P2</span><span class="op">)</span></span></code></pre></div>
<p><code>P1</code> and <code>P2</code> are the column names for the
parental genotypes. <code>filter_geno_by_parents</code> retains only
homozygous polymorphic markers between two specified parents.</p>
<p><strong>Note</strong>: The <code><a href="../reference/recode.html">recode()</a></code>function also
performs this filtering automatically. The use can specify also if
<code><a href="../reference/recode.html">recode()</a></code>should keep het markers.</p>
</div>
<div class="section level2">
<h2 id="recode-genotypes">Recode Genotypes<a class="anchor" aria-label="anchor" href="#recode-genotypes"></a>
</h2>
<p><code><a href="../reference/recode.html">geneticMapR::recode()</a></code> is the heart of the package. This
powerful function phases or recodes genotype marker data based on two
parental references (parent1 and parent2).</p>
<p>Recoding markers is one of the key steps in genetic mapping. The
markers are coded in a way that the haplotypes coming from either parent
can be tracked in the F<sub>2</sub> progeny.</p>
<p><img src="Recode_files/figure-html/recode%20fig-1.png" width="960"></p>
<div class="section level3">
<h3 id="typical-representation">Typical representation<a class="anchor" aria-label="anchor" href="#typical-representation"></a>
</h3>
<p>In the figure above panel A is a typical representation of an
intercross to derive an F<sub>2</sub> mapping population. The parental
lines are usually shown as fully homozygous:</p>
<ul>
<li>P1.a carries <em>only</em> the reference allele (blue, dosage =
0).</li>
<li>P2.b carries <em>only</em> the alternate allele (orange, dosage =
2).</li>
</ul>
<p>The resulting F1 is heterozygous (green, dosage = 1) at all loci.</p>
<p>F<sub>2</sub> individuals show a segregating mix of genotypes (0, 1,
2). This representation is valid when one your parents is the
<em>reference genome used to call SNPs</em>. If that is the case then no
recoding needs to occur. However, that is a rather rare case.</p>
</div>
<div class="section level3">
<h3 id="alternative-representation">Alternative representation<a class="anchor" aria-label="anchor" href="#alternative-representation"></a>
</h3>
<p>Parental genotypes, are assumed to be homozygous but are fixed for
different alleles across loci, allowing for more complex and realistic
representations:</p>
<ul>
<li>P1.b and P2.b carry <em>both</em> homozygous dosage values (0, or
2). Still homozygous but fixed for different alleles.</li>
</ul>
</div>
<div class="section level3">
<h3 id="outcrossing-species">Outcrossing species<a class="anchor" aria-label="anchor" href="#outcrossing-species"></a>
</h3>
<p>Most often and in outcrossing species like carrot and table beets or
others with high heterozygosity, the (inbred) parents may have mixed
genotypes with not totally homozygous genotypes. This is because some
loci are lethal on either homozygous configuration.</p>
<ul>
<li>P1.c and P2.c carry <em>some</em> heterozygous markers (1) in a low
proportion, with homozygous dosage values of 0 or 2.</li>
</ul>
<p>F<sub>1s</sub> are mostly heterozygous, but due to genotyping error
or residual heterozygosity some dosages of 0 and 2 are typically
observed in these individuals</p>
</div>
</div>
<div class="section level2">
<h2 id="recode-function">Recode function<a class="anchor" aria-label="anchor" href="#recode-function"></a>
</h2>
<p><code><a href="../reference/recode.html">recode()</a></code> was created to help with the task of recoding
the markers so that they reflect a the “typical representation. The
function is qute flexible and helps the user:</p>
<ul>
<li>Drop markers where either of the parent is NA</li>
<li>Remove non-polymorphic markers automatically (both parents have the
same genotype)</li>
<li>If <code>handle_het_markers</code> = TRUE, allows parental
heterozygous markers to be kept.</li>
</ul>
<p>Designed to work for F<sub>2</sub> populations.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Retains only homozygous markers</span></span>
<span><span class="va">hom_phased_geno_1629</span> <span class="op">&lt;-</span> <span class="fu">geneticMapR</span><span class="fu">::</span><span class="fu"><a href="../reference/recode.html">recode</a></span><span class="op">(</span>geno <span class="op">=</span> <span class="va">filtered_geno</span>,</span>
<span>                                parent1 <span class="op">=</span> <span class="va">P1</span>,</span>
<span>                                parent2 <span class="op">=</span> <span class="va">P2</span>,</span>
<span>                                numeric_output <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                handle_het_markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Dropping markers with NA in at least one of the parents. Your genotype matrix contains missing parental genotypes.</span></span>
<span><span class="co">#&gt; Dropping non-polymorphic markers. Your genotype matrix contains markers that are identical in the two parents.</span></span>
<span></span>
<span><span class="co"># Allows for heterozygous markers</span></span>
<span><span class="va">het_phased_geno_1629</span> <span class="op">&lt;-</span> <span class="fu">geneticMapR</span><span class="fu">::</span><span class="fu"><a href="../reference/recode.html">recode</a></span><span class="op">(</span>geno <span class="op">=</span> <span class="va">filtered_geno</span>,</span>
<span>                               parent1 <span class="op">=</span> <span class="va">P1</span>,</span>
<span>                               parent2 <span class="op">=</span> <span class="va">P2</span>,</span>
<span>                               numeric_output <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                               handle_het_markers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Dropping markers with NA in at least one of the parents. Your genotype matrix contains missing parental genotypes.</span></span>
<span><span class="co">#&gt; Dropping non-polymorphic markers. Your genotype matrix contains markers that are identical in the two parents.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tradeoff-between-marker-type-and-genome-coverage">Tradeoff Between Marker Type and Genome Coverage<a class="anchor" aria-label="anchor" href="#tradeoff-between-marker-type-and-genome-coverage"></a>
</h2>
<p>In outcrossing species I have noticed a tradeoff between coverage and
the quantity of heterozygous markers.</p>
<p>Including heterozygous markers increases genome coverage compared to
using only homozygous markers, but comes with tradeoffs. While
heterozygous markers can improve coverage, especially in regions with
segregation distortion or deleterious alleles, they may complicate
downstream linkage group estimation due to their different segregation
patterns.</p>
<p>The following section illustrates this tradeoff between marker type
and genome coverage.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_coverage.html">plot_coverage</a></span><span class="op">(</span>map <span class="op">=</span> <span class="fu"><a href="../reference/extract_map.html">extract_map</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Homozygous markers only"</span><span class="op">)</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_coverage.html">plot_coverage</a></span><span class="op">(</span>map <span class="op">=</span> <span class="fu"><a href="../reference/extract_map.html">extract_map</a></span><span class="op">(</span><span class="va">het_phased_geno_1629</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Homozygous and heterozygous markers"</span><span class="op">)</span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_genotype_histogram.html">plot_genotype_histogram</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span><span class="op">)</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_genotype_histogram.html">plot_genotype_histogram</a></span><span class="op">(</span><span class="va">het_phased_geno_1629</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">a</span>, <span class="va">b</span>, <span class="va">c</span>, <span class="va">d</span>, </span>
<span>          nrow <span class="op">=</span> <span class="fl">2</span>,</span>
<span>          ncol <span class="op">=</span> <span class="fl">2</span></span>
<span>          <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 193 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 801 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/tradeoff%20marker%20type%20v%20coverage-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="genotype-frequencies-and-filtering">Genotype Frequencies and Filtering<a class="anchor" aria-label="anchor" href="#genotype-frequencies-and-filtering"></a>
</h2>
<p><code>geneticMapR</code> includes functions to estimate frequencies
like <code><a href="../reference/freq.html">freq()</a></code> in an easy way. The function allows to estimate
genotype frequencies for later filtering.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Get Frequencies</span></span>
<span><span class="va">geno.freq.hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freq.html">freq</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span>,  input_format <span class="op">=</span> <span class="st">"numeric"</span>, by <span class="op">=</span><span class="st">"markers"</span><span class="op">)</span> </span>
<span><span class="va">geno.freq.het</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freq.html">freq</a></span><span class="op">(</span><span class="va">het_phased_geno_1629</span>, input_format <span class="op">=</span> <span class="st">"numeric"</span>, by <span class="op">=</span> <span class="st">"markers"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plots</span></span>
<span><span class="va">het_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/frequency_plot.html">frequency_plot</a></span><span class="op">(</span><span class="va">geno.freq.het</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Het markers, unfiltered"</span><span class="op">)</span></span>
<span><span class="va">hom_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/frequency_plot.html">frequency_plot</a></span><span class="op">(</span><span class="va">geno.freq.hom</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Hom markers, unfiltered"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Arrange</span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">hom_plot</span>,</span>
<span>          <span class="va">het_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/hom%20v%20het%20plot-1.png" width="768" style="display: block; margin: auto;"></p>
<p>From the plots above we see that there is an abundance of markers at
frequencies of 0 and 1 across all gentoypes (i.e., 0, 1, 2). We can use
the function <code><a href="../reference/filter_geno_by_freq.html">filter_geno_by_freq()</a></code> to filter markers out by
maximum genotype frequency and keeping a range of heterozygous
frequencies, say no lower than 0.1 or hiher than 0.70. Which is
reasonable for an F<sub>2</sub> mapping populations.</p>
<p>The resulting plot shows a filtered marker set and the overabundance
of markers at frequencies of 0 and 1 have been filtered out.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Heterozygous marker curation</span></span>
<span><span class="va">het_phased_geno_1629_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_geno_by_freq.html">filter_geno_by_freq</a></span><span class="op">(</span><span class="va">het_phased_geno_1629</span>, </span>
<span>                                                 max_geno_freq <span class="op">=</span> <span class="fl">0.90</span>, </span>
<span>                                                 het_freq_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.70</span><span class="op">)</span> </span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a dataset of frequencies based on the filtered data</span></span>
<span><span class="va">geno.freq.het.filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freq.html">freq</a></span><span class="op">(</span><span class="va">het_phased_geno_1629_filt</span>, </span>
<span>                               input_format <span class="op">=</span> <span class="st">"numeric"</span>, </span>
<span>                               by <span class="op">=</span><span class="st">"markers"</span><span class="op">)</span> </span>
<span></span>
<span></span>
<span><span class="co"># Create plot</span></span>
<span><span class="fu"><a href="../reference/frequency_plot.html">frequency_plot</a></span><span class="op">(</span><span class="va">geno.freq.het.filtered</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Het markers, filtered"</span><span class="op">)</span></span>
<span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/freq%20filtering-1.png" width="585.6" style="display: block; margin: auto;"></p>
<p>Below we see that our filtering paramaters have left us with a
typical 1:2:1 segregation ratio even for markers that included
heterozygous markers that were properly filtered.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">v</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_genotype_histogram.html">plot_genotype_histogram</a></span><span class="op">(</span><span class="va">het_phased_geno_1629_filt</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Homozygous + filtered heterozygous markers"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">w</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/plot_genotype_histogram.html">plot_genotype_histogram</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Homozygous markers"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">v</span>, <span class="va">w</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 435 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 193 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span></code></pre></div>
<p><img src="Recode_files/figure-html/filtering%20comparison-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="save-results">Save Results<a class="anchor" aria-label="anchor" href="#save-results"></a>
</h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">hom_phased_geno_1629</span>,</span>
<span>     <span class="va">het_phased_geno_1629</span>,</span>
<span>     <span class="va">het_phased_geno_1629_filt</span>,</span>
<span>     file <span class="op">=</span> <span class="st">"processed_data/R_data/filtered_geno_matrices_1629.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>A lot of genetic map construction revolves around the type of markers
you have and how you make good use of them. Very many filtering steps
are taken specially using GBS data to make sure the curated set of
markers are markers you can trust and informative.</p>
<p>This vignette provides a workflow for preparing and phasing genotypic
data for genetic mapping analyses in an F<sub>2</sub> population using
tools from <code>geneticMapR</code> and other bioinformatics packages.
Let’s continue in the next article the genetic map construction.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [8] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] png_0.1-8                   ggpubr_0.6.0               </span></span>
<span><span class="co">#&gt;  [3] ggplot2_3.5.2               tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [5] tidyr_1.3.1                 dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [7] VariantAnnotation_1.54.0    Rsamtools_2.24.0           </span></span>
<span><span class="co">#&gt;  [9] Biostrings_2.76.0           XVector_0.48.0             </span></span>
<span><span class="co">#&gt; [11] SummarizedExperiment_1.38.0 Biobase_2.68.0             </span></span>
<span><span class="co">#&gt; [13] GenomicRanges_1.60.0        GenomeInfoDb_1.44.0        </span></span>
<span><span class="co">#&gt; [15] IRanges_2.42.0              S4Vectors_0.46.0           </span></span>
<span><span class="co">#&gt; [17] MatrixGenerics_1.20.0       matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [19] BiocGenerics_0.54.0         generics_0.1.3             </span></span>
<span><span class="co">#&gt; [21] MapRtools_0.36              geneticMapR_0.0.0.9000     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_2.0.0           magrittr_2.0.3           GenomicFeatures_1.60.0  </span></span>
<span><span class="co">#&gt;   [4] farver_2.1.2             rmarkdown_2.29           BiocIO_1.18.0           </span></span>
<span><span class="co">#&gt;   [7] fs_1.6.6                 ragg_1.4.0               vctrs_0.6.5             </span></span>
<span><span class="co">#&gt;  [10] splines2_0.5.4           memoise_2.0.1            RCurl_1.98-1.17         </span></span>
<span><span class="co">#&gt;  [13] rstatix_0.7.2            htmltools_0.5.8.1        S4Arrays_1.8.0          </span></span>
<span><span class="co">#&gt;  [16] usethis_3.1.0            distributional_0.5.0     curl_6.2.2              </span></span>
<span><span class="co">#&gt;  [19] broom_1.0.8              SparseArray_1.8.0        Formula_1.2-5           </span></span>
<span><span class="co">#&gt;  [22] sass_0.4.10              bslib_0.9.0              htmlwidgets_1.6.4       </span></span>
<span><span class="co">#&gt;  [25] desc_1.4.3               cachem_1.1.0             GenomicAlignments_1.44.0</span></span>
<span><span class="co">#&gt;  [28] mime_0.13                lifecycle_1.0.4          iterators_1.0.14        </span></span>
<span><span class="co">#&gt;  [31] pkgconfig_2.0.3          Matrix_1.7-3             R6_2.6.1                </span></span>
<span><span class="co">#&gt;  [34] fastmap_1.2.0            GenomeInfoDbData_1.2.14  shiny_1.10.0            </span></span>
<span><span class="co">#&gt;  [37] digest_0.6.37            colorspace_2.1-1         AnnotationDbi_1.70.0    </span></span>
<span><span class="co">#&gt;  [40] ps_1.9.1                 pkgload_1.4.0            RSQLite_2.3.9           </span></span>
<span><span class="co">#&gt;  [43] textshaping_1.0.0        seriation_1.5.7          labeling_0.4.3          </span></span>
<span><span class="co">#&gt;  [46] httr_1.4.7               abind_1.4-8              mgcv_1.9-1              </span></span>
<span><span class="co">#&gt;  [49] compiler_4.5.0           remotes_2.5.0            withr_3.0.2             </span></span>
<span><span class="co">#&gt;  [52] bit64_4.6.0-1            HMM_1.0.1                backports_1.5.0         </span></span>
<span><span class="co">#&gt;  [55] BiocParallel_1.42.0      DBI_1.2.3                carData_3.0-5           </span></span>
<span><span class="co">#&gt;  [58] pkgbuild_1.4.7           ggsignif_0.6.4           DelayedArray_0.34.0     </span></span>
<span><span class="co">#&gt;  [61] sessioninfo_1.2.3        rjson_0.2.23             CVXR_1.0-15             </span></span>
<span><span class="co">#&gt;  [64] tools_4.5.0              httpuv_1.6.16            glue_1.8.0              </span></span>
<span><span class="co">#&gt;  [67] restfulr_0.0.15          callr_3.7.6              nlme_3.1-168            </span></span>
<span><span class="co">#&gt;  [70] promises_1.3.2           BSgenome_1.76.0          gtable_0.3.6            </span></span>
<span><span class="co">#&gt;  [73] qtl_1.70                 ca_0.71.1                car_3.1-3               </span></span>
<span><span class="co">#&gt;  [76] foreach_1.5.2            pillar_1.10.2            ggdist_3.3.2            </span></span>
<span><span class="co">#&gt;  [79] stringr_1.5.1            later_1.4.2              splines_4.5.0           </span></span>
<span><span class="co">#&gt;  [82] lattice_0.22-6           rtracklayer_1.68.0       gmp_0.7-5               </span></span>
<span><span class="co">#&gt;  [85] bit_4.6.0                tidyselect_1.2.1         registry_0.5-1          </span></span>
<span><span class="co">#&gt;  [88] miniUI_0.1.2             knitr_1.50               xfun_0.52               </span></span>
<span><span class="co">#&gt;  [91] devtools_2.4.5           scam_1.2-18              stringi_1.8.7           </span></span>
<span><span class="co">#&gt;  [94] UCSC.utils_1.4.0         yaml_2.3.10              evaluate_1.0.3          </span></span>
<span><span class="co">#&gt;  [97] codetools_0.2-20         BiocManager_1.30.25      cli_3.6.4               </span></span>
<span><span class="co">#&gt; [100] xtable_1.8-4             systemfonts_1.2.2        munsell_0.5.1           </span></span>
<span><span class="co">#&gt; [103] processx_3.8.6           jquerylib_0.1.4          Rcpp_1.0.14             </span></span>
<span><span class="co">#&gt; [106] XML_3.99-0.18            parallel_4.5.0           ellipsis_0.3.2          </span></span>
<span><span class="co">#&gt; [109] blob_1.2.4               pkgdown_2.1.1            profvis_0.4.0           </span></span>
<span><span class="co">#&gt; [112] urlchecker_1.0.1         bitops_1.0-9             Rmpfr_1.0-0             </span></span>
<span><span class="co">#&gt; [115] scales_1.3.0             purrr_1.0.4              crayon_1.5.3            </span></span>
<span><span class="co">#&gt; [118] rlang_1.1.6              cowplot_1.1.3            KEGGREST_1.48.0         </span></span>
<span><span class="co">#&gt; [121] TSP_1.2-4</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrey Vega Alfaro.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
